<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.43">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>5&nbsp; In Silico Model of CaMKII, NMDARs, and Associated Signalling Molecules – CaMKII-NMDAR interactions in Learning and Memory: A Case Study in Ethical and Reproducible Modelling Approaches</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./40-model-validation.html" rel="next">
<link href="./20-data-hazards.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-d4d76bf8491c20bad77d141916dc28e1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-0a0063f14e87237348a3db0020d9102a.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./30-model.html"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">In Silico Model of CaMKII, NMDARs, and Associated Signalling Molecules</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">CaMKII-NMDAR interactions in Learning and Memory: A Case Study in Ethical and Reproducible Modelling Approaches</a> 
        <div class="sidebar-tools-main">
    <a href="./CaMKII-NMDAR-interactions-in-Learning-and-Memory--A-Case-Study-in-Ethical-and-Reproducible-Modelling-Approaches.pdf" title="Download PDF" class="quarto-navigation-tool px-1" aria-label="Download PDF"><i class="bi bi-file-pdf"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Declaration</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./10-intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introduction</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./20-intro-biology.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Biological Research Background</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./30-modelling_background.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Computational Modelling Background</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./20-repro_open.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Open, Reproducible and Ethics-focused PhD</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./20-data-hazards.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">A Case Study in Ethical Reflection: Data Hazards in this PhD</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./30-model.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">In Silico Model of CaMKII, NMDARs, and Associated Signalling Molecules</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./40-model-validation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Validating model behaviour</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./40-results.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Analysis of CaMKII Dynamics in Wild-Type and Mutant Models</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./publications.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">List of publications and presentations resulting from this PhD</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#previous-work" id="toc-previous-work" class="nav-link active" data-scroll-target="#previous-work"><span class="header-section-number">5.1</span> Previous work</a></li>
  <li><a href="#model-description" id="toc-model-description" class="nav-link" data-scroll-target="#model-description"><span class="header-section-number">5.2</span> Model Description</a>
  <ul class="collapse">
  <li><a href="#molecule-concentrations-localization-and-cell-volume" id="toc-molecule-concentrations-localization-and-cell-volume" class="nav-link" data-scroll-target="#molecule-concentrations-localization-and-cell-volume"><span class="header-section-number">5.2.1</span> Molecule concentrations, localization and cell volume</a></li>
  <li><a href="#sec-camkii-as-dodecamer" id="toc-sec-camkii-as-dodecamer" class="nav-link" data-scroll-target="#sec-camkii-as-dodecamer"><span class="header-section-number">5.2.2</span> Modelling CaMKII as a dodecamer</a></li>
  <li><a href="#camkii-and-nmdar-binding" id="toc-camkii-and-nmdar-binding" class="nav-link" data-scroll-target="#camkii-and-nmdar-binding"><span class="header-section-number">5.2.3</span> CaMKII and NMDAR binding</a></li>
  <li><a href="#sec-binding-ca-cam" id="toc-sec-binding-ca-cam" class="nav-link" data-scroll-target="#sec-binding-ca-cam"><span class="header-section-number">5.2.4</span> Binding of calcium to calmodulin</a></li>
  <li><a href="#sec-camkii-flicker" id="toc-sec-camkii-flicker" class="nav-link" data-scroll-target="#sec-camkii-flicker"><span class="header-section-number">5.2.5</span> CaMKII subunits flicker between open and closed states</a></li>
  <li><a href="#sec-binding-cam-camkii" id="toc-sec-binding-cam-camkii" class="nav-link" data-scroll-target="#sec-binding-cam-camkii"><span class="header-section-number">5.2.6</span> Binding of CaM to CaMKII</a></li>
  <li><a href="#camkii-phosphorylation-and-dephosphorylation" id="toc-camkii-phosphorylation-and-dephosphorylation" class="nav-link" data-scroll-target="#camkii-phosphorylation-and-dephosphorylation"><span class="header-section-number">5.2.7</span> CaMKII phosphorylation and dephosphorylation</a></li>
  <li><a href="#t286-phosphorylation" id="toc-t286-phosphorylation" class="nav-link" data-scroll-target="#t286-phosphorylation"><span class="header-section-number">5.2.8</span> T286 phosphorylation</a></li>
  <li><a href="#t306-phosphorylation" id="toc-t306-phosphorylation" class="nav-link" data-scroll-target="#t306-phosphorylation"><span class="header-section-number">5.2.9</span> T306 phosphorylation</a></li>
  <li><a href="#dephosphorylation-of-camkii-subunits" id="toc-dephosphorylation-of-camkii-subunits" class="nav-link" data-scroll-target="#dephosphorylation-of-camkii-subunits"><span class="header-section-number">5.2.10</span> Dephosphorylation of CaMKII subunits</a></li>
  <li><a href="#reaction-rules-and-kinetic-rates" id="toc-reaction-rules-and-kinetic-rates" class="nav-link" data-scroll-target="#reaction-rules-and-kinetic-rates"><span class="header-section-number">5.2.11</span> Reaction rules and kinetic rates</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span id="sec-model-description" class="quarto-section-identifier"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">In Silico Model of CaMKII, NMDARs, and Associated Signalling Molecules</span></span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>Numerous computational models of CaMKII dynamics have been developed over the past decade to look at different interactions of CaMKII and its substrates or binding partners at varying levels of detail <span class="citation" data-cites="holmes2000Models lisman2001Model lisman2002Molecular zhabotinsky2000Bistability lucic2008Detaileda stefan2008Allostericd stefan2011MultiStage johnson2015Model pharris2019Multistate ordyan2020Interactions bartol2024Spatial">(<a href="#ref-holmes2000Models" role="doc-biblioref">Holmes 2000</a>; <a href="#ref-lisman2001Model" role="doc-biblioref">J. E. Lisman and Zhabotinsky 2001</a>; <a href="#ref-lisman2002Molecular" role="doc-biblioref">J. Lisman, Schulman, and Cline 2002</a>; <a href="#ref-zhabotinsky2000Bistability" role="doc-biblioref">Zhabotinsky 2000</a>; <a href="#ref-lucic2008Detaileda" role="doc-biblioref">Lučić, Greif, and Kennedy 2008</a>; <a href="#ref-stefan2008Allostericd" role="doc-biblioref">M. I. Stefan, Edelstein, and Le Novère 2008</a>; <a href="#ref-stefan2011MultiStage" role="doc-biblioref">M. Stefan et al. 2011</a>; <a href="#ref-johnson2015Model" role="doc-biblioref">Johnson et al. 2015</a>; <a href="#ref-pharris2019Multistate" role="doc-biblioref">Pharris et al. 2019</a>; <a href="#ref-ordyan2020Interactions" role="doc-biblioref">Ordyan et al. 2020</a>; <a href="#ref-bartol2024Spatial" role="doc-biblioref">Bartol et al. 2024</a>)</span>. Many of these models center on the binding of calcium to calmodulin (CaM) and the subsequent formation of the CaM-CaMKII complex. While these studies provide a good foundation for understanding CaMKII dynamics, the models either focus solely on temporal dynamics without considering geometry, or explore both spatial and temporal dynamics but do not model CaMKII as a multimeric molecule. Some models do account for CaMKII as a multimeric dodecamer <span class="citation" data-cites="pharris2019Multistate ordyan2020Interactions bartol2024Spatial">(<a href="#ref-pharris2019Multistate" role="doc-biblioref">Pharris et al. 2019</a>; <a href="#ref-ordyan2020Interactions" role="doc-biblioref">Ordyan et al. 2020</a>; <a href="#ref-bartol2024Spatial" role="doc-biblioref">Bartol et al. 2024</a>)</span> but do not focus on its interactions with NMDARs. To my knowledge, this research is novel in the field, as it incorporates all these aspects: modeling CaMKII as a dodecamer, examining spatiotemporal dynamics, and focusing specifically on the CaMKII/NMDAR complex.</p>
<p>The research carried out in this PhD project follows recent studies <span class="citation" data-cites="pharris2019Multistate ordyan2020Interactions bartol2024Spatial">(<a href="#ref-pharris2019Multistate" role="doc-biblioref">Pharris et al. 2019</a>; <a href="#ref-ordyan2020Interactions" role="doc-biblioref">Ordyan et al. 2020</a>; <a href="#ref-bartol2024Spatial" role="doc-biblioref">Bartol et al. 2024</a>)</span> that have looked at CaMKII as a dodecamer to study its autophosphorylation and spatiotemporal regulation in poststynaptic dendrites using BioNetGen and MCell. Incorporating spatial, temporal, and multimeric aspects is crucial because these dimensions fundamentally shape the behavior of molecular complexes like like the CaMKII/NMDAR interaction. Spatial dynamics can reveal how the molecules interact within different cellular regions; CaMKII phosphorylation levels, as well as its functions have been shown to differ depending on its subcellular localization <span class="citation" data-cites="davies2007ACaMKII">(<a href="#ref-davies2007ACaMKII" role="doc-biblioref">Davies et al. 2007</a>)</span>. Temporal dynamics are important to understand how CaMKII/NMDAR interactions evolve over time. The duration and timing of CaMKII activation influence whether synaptic changes are transient or long-lasting, affecting the induction and maintenance of LTP and LTD.</p>
<section id="previous-work" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="previous-work"><span class="header-section-number">5.1</span> Previous work</h2>
<p>Several multi-state models of CaMKII have been proposed, each differing in focus and scope from the present one <span class="citation" data-cites="holmes2000Models lisman2001Model lisman2002Molecular zhabotinsky2000Bistability lucic2008Detaileda stefan2008Allostericd stefan2011MultiStage johnson2015Model pharris2019Multistate ordyan2020Interactions bartol2024Spatial">(<a href="#ref-holmes2000Models" role="doc-biblioref">Holmes 2000</a>; <a href="#ref-lisman2001Model" role="doc-biblioref">J. E. Lisman and Zhabotinsky 2001</a>; <a href="#ref-lisman2002Molecular" role="doc-biblioref">J. Lisman, Schulman, and Cline 2002</a>; <a href="#ref-zhabotinsky2000Bistability" role="doc-biblioref">Zhabotinsky 2000</a>; <a href="#ref-lucic2008Detaileda" role="doc-biblioref">Lučić, Greif, and Kennedy 2008</a>; <a href="#ref-stefan2008Allostericd" role="doc-biblioref">M. I. Stefan, Edelstein, and Le Novère 2008</a>; <a href="#ref-stefan2011MultiStage" role="doc-biblioref">M. Stefan et al. 2011</a>; <a href="#ref-johnson2015Model" role="doc-biblioref">Johnson et al. 2015</a>; <a href="#ref-pharris2019Multistate" role="doc-biblioref">Pharris et al. 2019</a>; <a href="#ref-ordyan2020Interactions" role="doc-biblioref">Ordyan et al. 2020</a>; <a href="#ref-bartol2024Spatial" role="doc-biblioref">Bartol et al. 2024</a>)</span>. Notably, there are multiple models by Stefan et al. <span class="citation" data-cites="stefan2008Allostericd stefan2011MultiStage">(<a href="#ref-stefan2008Allostericd" role="doc-biblioref">M. I. Stefan, Edelstein, and Le Novère 2008</a>; <a href="#ref-stefan2011MultiStage" role="doc-biblioref">M. Stefan et al. 2011</a>)</span> on CaMKII which explore the dynamics between CaMKII and CaM using Stochsim simulator <span class="citation" data-cites="lenovere2001StochSim">(<a href="#ref-lenovere2001StochSim" role="doc-biblioref">Le Novère and Shimizu 2001</a>)</span> which does not does not account for spatial effects, unlike MCell. These computational models provide insights into the mechanisms of calcium binding to CaM and the subsequent interaction with CaMKII, influencing the kinase’s activation, phosphorylation, and opening and closing dynamics. Notably, <span class="citation" data-cites="pharris2019Multistate">Pharris et al. (<a href="#ref-pharris2019Multistate" role="doc-biblioref">2019</a>)</span> published a model that specifically examines the dynamics of CaMKII in relation to CaM using MCell. However, this study was conducted using MCell 3.3, which did not support the diffusion of multi-state molecules. In this model, CaMKII was represented as a dodecamer, though not using BNGL, and the holoenzymes themselves were non-diffusing. Instead, the authors implemented a model in which CaM could diffuse and interact with a non-diffusing multi-state representation of CaMKII.</p>
<p><span class="citation" data-cites="ordyan2020Interactions">Ordyan et al. (<a href="#ref-ordyan2020Interactions" role="doc-biblioref">2020</a>)</span> employed a BioNetGen rule-based modelling approach using Monte Carlo methods, similar to the present study, to investigate how <span class="math inline">\(\mathrm{Ca^{2+}}\)</span> signals influence CaMKII phosphorylation dynamics within the PSD. Their work primarily focuses on the interactions between CaM and the scaffolding molecule Neurogranin (Ng) in regulating CaMKII dynamics. However, their study does not investigate the crucial interaction between CaMKII and NMDARs.</p>
<p>More recently, <span class="citation" data-cites="bartol2024Spatial">Bartol et al. (<a href="#ref-bartol2024Spatial" role="doc-biblioref">2024</a>)</span> have published a study using the same version of MCell 4 as the model used in this PhD. Theyr use particle-based stochastic simulations in MCell4 to model the activation of CaMKII by <span class="math inline">\(\mathrm{Ca^{2+}}\)</span> flowing through NMDARs within a postsynaptic spine. Their spatiotemporal model simulates CaMKII as a dodecamer with calcium influx through NMDARs, yet they do not directly examine interactions between NMDARs and CaMKII.</p>
<p>Each of these models represents a step towards the development of the present study, which advances the field by explicitly modelling CaMKII as a dodecamer that directly interacts with NMDARs on the surface of the postsynaptic dendrite.</p>
</section>
<section id="model-description" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="model-description"><span class="header-section-number">5.2</span> Model Description</h2>
<p>In this model, CaMKII is represented as a dodecamer using BioNetGen <a href="#sec-camkii-as-dodecamer" class="quarto-xref"><span class="quarto-unresolved-ref">sec-camkii-as-dodecamer</span></a>. The model simulates its diffusion over time and space within a postsynaptic density (PSD) volume. Other molecules involved in the system include calcium, CaM, and protein phosphatase 1 (PP1), and NMDARs. The following sections provide a detailed breakdown of how these reactions are modelled, highlighting their biological relevance.</p>
<section id="molecule-concentrations-localization-and-cell-volume" class="level3" data-number="5.2.1">
<h3 data-number="5.2.1" class="anchored" data-anchor-id="molecule-concentrations-localization-and-cell-volume"><span class="header-section-number">5.2.1</span> Molecule concentrations, localization and cell volume</h3>
<p>All molecules (except NMDARs which are surface molecules) are released from inside the cell to interact with each other. Molecules are released as discrete molecule numbers. Concentrations can be calculated as shown below in <a href="#eq-molec-concentrations-calc" class="quarto-xref">Equation&nbsp;<span class="quarto-unresolved-ref">eq-molec-concentrations-calc</span></a>, where we begin with the number of molecules <span class="math inline">\(N\)</span> and convert it to moles using Avogadro’s number <span class="math inline">\(N_A\)</span>. Next, we account for the volume in cubic micrometers, represented as <span class="math inline">\(V \, (\mu \text{m}^3)\)</span>. This equation gives us concentration in Molar.</p>
<p><span id="eq-molec-concentrations-calc"><span class="math display">\[
N \, \text{molecules} \times
\frac{1 \, \text{mol}}{N_A \, \text{molecules}} \times
\frac{1}{V(\mu \text{m}^3)} \times
\frac{V \, (\mu \text{m}^3)}{V \, (\text{L})} =
C_1 \, (\text{M})
\tag{5.1}\]</span></span></p>
<p>The example calculation below considers an initial release of 10000 molecules and a dendritic volume of 0.5 µm³ (<a href="#eq-molec-concentrations-calc-example" class="quarto-xref">Equation&nbsp;<span class="quarto-unresolved-ref">eq-molec-concentrations-calc-example</span></a>). <span id="eq-molec-concentrations-calc-example"><span class="math display">\[
1*10^{4} \text{molecules} \times
\frac{1 mol}{6.023*10^{23} \text{molecules}} \times
\frac{1}{0.5 \, \mu \text{m}^3} \times
\frac{1  \, \mu \text{m}^3}{10 ^{-15} \text{L}} =
3.32*10^{-6} \text{M}
\tag{5.2}\]</span></span></p>
<p>In the model presented here, reactions of cytoplasmatic molecules specified in BNGL, were placed into the geometry of a postysnaptic dendritic spine with a volume of 0.50588 <span class="math inline">\(\mu\text{m}^3\)</span>, which matches the volume chosen by <span class="citation" data-cites="ordyan2020Interactions">Ordyan et al. (<a href="#ref-ordyan2020Interactions" role="doc-biblioref">2020</a>)</span> in their model of CaMKII in the PSD and is within ranges of spine volume of 0.004 to 0.6 μm3 in hippocampal CA1 neurons (Harris et al., 1989). Moreover, Tonnesen et al.&nbsp;(2018) report a median spine volume of 0.1 <span class="math inline">\(\mu\text{m}^3\)</span>, with some spines reaching up to 0.5~<span class="math inline">\(\mu\text{m}^3\)</span>. As such, a volume of 0.5 <span class="math inline">\(\mu\text{m}^3\)</span> is within physiologically observed bounds, albeit on the higher end of the range.</p>
<p>The molecular concentrations used in this model fall within physiologically plausible ranges reported in the literature. For example, the initial release of 1000 <span class="math inline">\(\mathrm{Ca^{2+}}\)</span> ions corresponds to a concentration of 3.28 <span class="math inline">\(\mu\text{M}\)</span>. Prior studies employing fluorescent <span class="math inline">\(\mathrm{Ca^{2+}}\)</span> indicators have shown resting intracellular <span class="math inline">\(\mathrm{Ca^{2+}}\)</span> concentrations ([<span class="math inline">\(\mathrm{Ca^{2+}}\)</span>]<span class="math inline">\(_i\)</span>) in the range of 0.05–0.1 <span class="math inline">\(\mu\text{M}\)</span>, with increases up to 100-fold under stimulated conditions (Grienberger et al., 2012; Maravall et al., 2000). As such, the calcium input concentrations used in this model are consistent with the expected values under simulated conditions. Similarly, while the concentration of calmodulin (CaM) can vary widely in different studies, ranging from 0.1 <span class="math inline">\(\mu\text{M}\)</span> to 8 <span class="math inline">\(\mu\text{M}\)</span>, with some reports extending up to 250 <span class="math inline">\(\mu\text{M}\)</span>, the concentration of 0.98 <span class="math inline">\(\mu\text{M}\)</span> used in this model falls within these reported ranges. A more in depth discussion of the validity of chosen values and potential limitations is provided in <a href="#sec-validating-model" class="quarto-xref"><span class="quarto-unresolved-ref">sec-validating-model</span></a>.</p>
<p>CaMKII is modelled as a dodecameric molecule, which is biologically accurate. The model contains four protein species: CaM, PP1, CaMKII and NMDARS (see table of molecules included in the model). Initial molecules released inside the cell are: <span class="math inline">\(\mathrm{Ca^{2+}}\)</span> ions, CaM, CaMKII and PP1, inside the cytosol, and NMDARs as cell surface molecules. Since four calcium ions are needed to fully saturate CaM (CaM_Ca4) [REFERENCES], THIS IS NOT TRUE. REFERENCE - CHECK. HELP. five times more calcium than CaM and CaMKII was released to ensure it would not be a limiting factor [TABLE REFERENCE]. An insufficient amount of calcium would limit the formation of fully active CaM, thereby reducing its ability to activate CaMKII. This could introduce an unintended bottleneck in the signalling cascade, making it difficult to accurately study CaMKII activation, autophosphorylation, or its downstream effects. Interactions between different molecules are modelled as rules in BNGL (as explained in CHAPTER REFERENCE), where each molecule reacts according to certain conditions and kinetic rates.</p>
<div id="fig-molecule-concentrations" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="Molecule concentrations in the model" data-fig-pos="H">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-molecule-concentrations-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="30-methods-figures\molecule-concentrations.PNG" class="img-fluid figure-img" data-fig-scap="Molecule concentrations in the model" data-fig-pos="H">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-molecule-concentrations-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5.1: Molecules released into the model, including their defined possible states specified in BioNetGen. Concentrations were calculated using Equation <a href="#eq-molec-concentrations-calc" class="quarto-xref">Equation&nbsp;<span class="quarto-unresolved-ref">eq-molec-concentrations-calc</span></a>, based on a simulation volume of 0.50588 <span class="math inline">\(\mu\text{m}^3\)</span>. *Note: 60 CaMKII holoenzymes, each comprising 12 subunits, correspond to a total of 720 subunits (60 × 12).
</figcaption>
</figure>
</div>
</section>
<section id="sec-camkii-as-dodecamer" class="level3" data-number="5.2.2">
<h3 data-number="5.2.2" class="anchored" data-anchor-id="sec-camkii-as-dodecamer"><span class="header-section-number">5.2.2</span> Modelling CaMKII as a dodecamer</h3>
<p>In the model at hand, the individual subunits are specified in BNGL as follows: <em>CaMKII(open<sub>0</sub>1,T286<sub>0</sub>P,T306<sub>0</sub>P,cam,nmdar,l,r,c)</em>. The components l, r, and c represent binding sites within the holoenzyme: l is a binding site on the left of the subunit, r on the right, and c is the central site interacting with an opposing subunit in the six-membered ring (<a href="#fig-camkii-dodecamer" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-camkii-dodecamer</span></a>). In BNGL syntax, interactions between components are represented by a period, with a “!” (bang) followed by a number to denote specific binding. For example, a bond between the right side of one subunit and the left side of another is written as shown in <a href="#fig-camkii-dodecamer" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-camkii-dodecamer</span></a> (B). The “!” is followed by any number, provided the two binding components share the same number.</p>
<p>Modelling CaMKII as a dodecamer using BNGL enables the representation of the molecule as multimeric, with each subunit able to adopt the following possible states:</p>
<ul>
<li>A CaM binding site, CaM can be bound or unbound,</li>
<li>A NMDAR binding site, CaMKII can be bound or unbound to NMDARs,</li>
<li>A T286 phosphorylation site, which can be unphosphorylated (T286~0) or phosphorylated (T286~P),</li>
<li>A T306 phosphorylation site, which can be unphosphorylated (T306~0) or phosphorylated (T306~P),</li>
<li>A status of being open (open~1) or closed (open~0).</li>
</ul>
<div id="fig-camkii-dodecamer" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="CaMKII dodecamer in BNGL">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-camkii-dodecamer-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="30-methods-figures\camkii_dodecamer_bngl.PNG" class="img-fluid figure-img" style="width:95.0%" data-fig-scap="CaMKII dodecamer in BNGL">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-camkii-dodecamer-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5.2: (A) The open flag determines whether the subunit is in a closed (0) or open (1) state. The T286 and T306 flags represent the phosphorylation states, where a value of 0 indicates the subunit is unphosphorylated, and “P” indicates phosphorylation. The cam flag refers to a binding site for CaM, which can either be bound or unbound. Similarly, the nmdar flag represents another binding site for the NMDA receptor, which can also be in a bound or unbound state. The cam and nmdar flags represent binding events that can occur or not, depending on whether CaM or NMDA receptors are bound to the subunit. (B) (B) The left (l), right (r), and centre (c) binding sites facilitate the assembly of subunits into two hexameric rings. These rings are stabilised via interactions at the centre binding site. Subunits participating in the left and right rings are highlighted in matching colours to distinguish the respective assemblies. (C)A more visual depiction of the two separated hexameric rings. (D) The centre binding interactions are specifically highlighted to show the dodecameric representation of CaMKII in the model.
</figcaption>
</figure>
</div>
<p>The model includes autonomous activation of CaMKII subunits, allowing them to flicker between open and closed states when unbound to CaM and unphosphorylated at T286 (see reaction rule #5 in <a href="#fig-tbl-1" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-tbl-1</span></a>). The binding of CaM or phosphorylation at T286 stabilises the open state by preventing reversion to a closed conformation. Additionally, CaM binding can only occur if the subunit is in an open state due to the structural requirements for high-affinity binding. Electron microscopy of rat <span class="math inline">\(\alpha\)</span>-CaMKII in Sf9 cells <span class="citation" data-cites="myers2017CaMKII">(<a href="#ref-myers2017CaMKII" role="doc-biblioref">Myers et al. 2017</a>)</span> suggests that fewer than 3% of CaMKII subunits adopt a compact, docked-like conformation <span class="citation" data-cites="bayer2019CaM">(<a href="#ref-bayer2019CaM" role="doc-biblioref">K. Ulrich Bayer and Schulman 2019</a>)</span>. The model therefore assumes all CaMKII to be in an undocked confirmation, in order to streamline the focus on the fundamental functional states of CaMKII.</p>
<p>Each CaMKII holoenzyme diffuses through the cell volume with a specified diffusion constant of <span class="math inline">\(1\times 10^{-6} \text{cm}^2/\text{s}\)</span>, which is within biological limits <span class="citation" data-cites="sanabria2008Spatial">(<a href="#ref-sanabria2008Spatial" role="doc-biblioref">Sanabria et al. 2008</a>)</span>. In the model, CaMKII subunits can interact with NMDA receptors located on the cell surface. When they interact, CaMKII can bind reversibly to the NMDA receptor, forming a complex in which the CaMKII subunit that has been bound remains open (see reaction rule #12 in <a href="#fig-tbl-1" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-tbl-1</span></a>). As CaMKII remains open when bound to NMDARs, further reactions can happen such as phosphorylation at T286 and T305, as well as binding to CaM molecules <a href="#fig-camkii_nmdar" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-camkii_nmdar</span></a>.</p>
</section>
<section id="camkii-and-nmdar-binding" class="level3" data-number="5.2.3">
<h3 data-number="5.2.3" class="anchored" data-anchor-id="camkii-and-nmdar-binding"><span class="header-section-number">5.2.3</span> CaMKII and NMDAR binding</h3>
<p>In this model, NMDARs are represented as single-site molecules, rather than as tetramers, due to the fact that the primary interaction between CaMKII and NMDARs occurs predominantly through the GluN2B subunit. While it is important to acknowledge that the GluN2B subunit is distinct from the entire NMDAR heteromer, we refer to the model as NMDAR for simplicity and clarity. This nomenclature is used as a practical approximation, since the interactions we are modelling are specifically associated with the GluN2B subunit, which is often studied in isolation in experimental settings. In many in vitro experiments, the focus is on the GluN2B subunit, and the behaviour of the whole NMDAR complex is inferred based on these more controlled interactions <span class="citation" data-cites="akashi2009NMDA goodell2014CaMKII barcomb2016CaMKIIa rumian2024LTP">(<a href="#ref-akashi2009NMDA" role="doc-biblioref">Akashi et al. 2009</a>; <a href="#ref-goodell2014CaMKII" role="doc-biblioref">Goodell et al. 2014</a>; <a href="#ref-barcomb2016CaMKIIa" role="doc-biblioref">Barcomb et al. 2016</a>; <a href="#ref-rumian2024LTP" role="doc-biblioref">Rumian et al. 2024</a>)</span>. Therefore, for the purposes of this model, referring to the system as NMDAR aligns with the established experimental approach, even though the underlying interactions are predominantly occurring via the GluN2B subunit. Further developments of this model to account for a more complex NMDAR molecule are possible, as I propose in <a href="#sec-future-directions" class="quarto-xref"><span class="quarto-unresolved-ref">sec-future-directions</span></a>.</p>
<div id="fig-camkii_nmdar" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="Simplified schematic of CaMKII binding to NMDARs in Compartmental BioNetGen" data-fig-pos="H">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-camkii_nmdar-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="30-methods-figures\cbngl_camkii_nmdar.PNG" class="img-fluid figure-img" style="width:85.0%" data-fig-scap="Simplified schematic of CaMKII binding to NMDARs in Compartmental BioNetGen" data-fig-pos="H">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-camkii_nmdar-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5.3: Simplified schematic of CaMKII binding to NMDARs in Compartmental BioNetGen. (A) A cell is defined with plasma membrane (PM) and cytoplasm (CP) volumes. Molecules are released as surface molecules (NMDARs) or volume molecules (rest). (B) NMDARs embedded in the PM can bind to CaMKII subunits diffusing within the cytoplasm. (C) Different subunits within the CaMKII dodecamer (simplified here as a hexamer) can independently interact with other molecules such as NMDAR or CaM. These interactions can occur simultaneously or asynchronously, allowing CaMKII to integrate multiple signals through subunit-specific states.
</figcaption>
</figure>
</div>
<p>Furthermore, as previously discussed in <a href="#sec-CaMKII-NMDAR-association" class="quarto-xref"><span class="quarto-unresolved-ref">sec-CaMKII-NMDAR-association</span></a>, the precise molecular mechanism by which CaMKII binds to NMDARs is not yet fully understood. Nonetheless, the literature seems to agree that CaMKII interaction with the C-terminal domain (CTD) of NMDARs plays a key role in stabilising CaMKII in its open, catalytically active conformation <span class="citation" data-cites="bayer2001Interaction bayer2006Transitionb ozden2022CaMKIIb nicoll2023Synaptic">(<a href="#ref-bayer2001Interaction" role="doc-biblioref">K.-Ulrich Bayer et al. 2001</a>; <a href="#ref-bayer2006Transitionb" role="doc-biblioref">K. Ulrich Bayer et al. 2006</a>; <a href="#ref-ozden2022CaMKIIb" role="doc-biblioref">Özden et al. 2022</a>; <a href="#ref-nicoll2023Synaptic" role="doc-biblioref">Nicoll and Schulman 2023</a>)</span>. In our model, CaMKII may bind to an NMDAR provided that it is in the open conformation and that the receptor possesses an available CaMKII-binding site (see reaction rule 12 in TABLE-REFERENCE). Upon binding, a stable CaMKII/NMDAR complex is formed, in which the kinase remains in its active state due to the stabilisation of its open structure.</p>
</section>
<section id="sec-binding-ca-cam" class="level3" data-number="5.2.4">
<h3 data-number="5.2.4" class="anchored" data-anchor-id="sec-binding-ca-cam"><span class="header-section-number">5.2.4</span> Binding of calcium to calmodulin</h3>
<p>CaM has four EF hands divided into N and C lobes capable of binding two calcium ions per lobe, consequently CaM is modelled as being able to bind four calcium ions in our model. The N lobe binds calcium faster with lower affinity and the C lobe binds slower with higher affinity <span class="citation" data-cites="kawasaki2017Conformational">(<a href="#ref-kawasaki2017Conformational" role="doc-biblioref">Kawasaki and Kretsinger 2017</a>)</span>. Numerous models have been developed to investigate the dynamics of calcium ions binding to CaM <span class="citation" data-cites="stefan2008Allostericd pepke2010Dynamic gonzalez-andrade2016Insights">(<a href="#ref-stefan2008Allostericd" role="doc-biblioref">M. I. Stefan, Edelstein, and Le Novère 2008</a>; <a href="#ref-pepke2010Dynamic" role="doc-biblioref">Pepke et al. 2010</a>; <a href="#ref-gonzalez-andrade2016Insights" role="doc-biblioref">González-Andrade et al. 2016</a>)</span>. <span class="citation" data-cites="linkevicius2025Fitting">Linkevicius et al. (<a href="#ref-linkevicius2025Fitting" role="doc-biblioref">2025</a>)</span> examined and compared various of these models, providing a useful summary of the typical categories for modeling CaM binding to <span class="math inline">\(\mathrm{Ca^{2+}}\)</span>. The model used here falls under their proposed “scheme 3” (<a href="#fig-scheme3-cam" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-scheme3-cam</span></a>), where calcium binds and unbinds to four separate calcium binding sites in CaM. Unlike other models that distinguish between CaM’s N-terminal and C-terminal binding lobes, this project does not, as the specifics of calcium binding to CaM are beyond its scope and following “scheme 3” approach to calcium binding produces biologically relevant results while avoiding complexity, allowing us to focus on further CaMKII and NMDARs interactions.</p>
<div id="fig-scheme3-cam" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="Calcium binding to CaM" data-fig-pos="H">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-scheme3-cam-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="30-methods-figures\scheme3_cam.PNG" class="img-fluid figure-img" style="width:70.0%" data-fig-scap="Calcium binding to CaM" data-fig-pos="H">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-scheme3-cam-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5.4: We model calcium binding to CaM following “scheme 3” proposed by Linkevicius et al., 2024. It is a sequential CaM scheme where each binding site is represented individually.
</figcaption>
</figure>
</div>
</section>
<section id="sec-camkii-flicker" class="level3" data-number="5.2.5">
<h3 data-number="5.2.5" class="anchored" data-anchor-id="sec-camkii-flicker"><span class="header-section-number">5.2.5</span> CaMKII subunits flicker between open and closed states</h3>
<p>It is well established that CaMKII subunits can exist in both open and closed conformations <span class="citation" data-cites="pullara2017Mechanisms zhang2017Opposing">(<a href="#ref-pullara2017Mechanisms" role="doc-biblioref">Pullara, Asciutto, and General 2017</a>; <a href="#ref-zhang2017Opposing" role="doc-biblioref">Zhang et al. 2017</a>)</span>, even in the absence of CaM <span class="citation" data-cites="asciutto2021Physicalc">(<a href="#ref-asciutto2021Physicalc" role="doc-biblioref">Asciutto, Pantano, and General 2021</a>)</span>. While the notion that CaM induces a conformational change in CaMKII is the common view, the assumption that this binding event directly triggers subunit activation is largely based on the induced-fit model, as discussed in <a href="#sec-cam-binding-and-t286-p" class="quarto-xref"><span class="quarto-unresolved-ref">sec-cam-binding-and-t286-p</span></a>. These observations may be due to the fact that binding of <span class="math inline">\(\mathrm{Ca^{2+}}\)</span>/CaM to CaMKII is more likely when the CaMKII kinase subunits are in the open conformation.</p>
<p>CaMKII subunits inherently fluctuate between different conformational states (open and closed), and an alternative perspective is that CaM does not directly induce the opening of the subunit but rather stabilises the open conformation upon binding <span class="citation" data-cites="asciutto2021Physicalc">(<a href="#ref-asciutto2021Physicalc" role="doc-biblioref">Asciutto, Pantano, and General 2021</a>)</span>. This suggests that the enzyme does not strictly require CaM to transition into an open state; rather, CaM serves to stabilise and prolong this conformation, thereby sustaining enzymatic activity. From a dynamic standpoint, it is plausible that, in the absence of CaM, there exists a low probability of the subunit adopting an open conformation due to intrinsic structural fluctuations. Upon CaM binding, this equilibrium may shift, favouring the stabilisation of the open state and ultimately leading to the full activation observed in vitro <span class="citation" data-cites="asciutto2021Physicalc">(<a href="#ref-asciutto2021Physicalc" role="doc-biblioref">Asciutto, Pantano, and General 2021</a>)</span>. Accordingly, in this model, CaMKII subunits are considered to undergo spontaneous transitions between open and closed conformations (see #a in <a href="#fig-camkii-binding-rxns" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-camkii-binding-rxns</span></a> for schematic depiction), and the impact of CaM binding is assessed in terms of its stabilising effects on these states.</p>
<p>CaM binding is not the only mechanism that can render CaMKII subunits in their open conformation. In the model used in this PhD, as per the known literature, CaMKII subunits can remain in an open conformation due to one or more of the following events:</p>
<ul>
<li>CaM binding: CaM binding has been shown to structurally stabilise the open conformation of the CaMKII kinase subunit, preventing the kinase domain from re-engaging with its own regulatory region, thereby maintaining the open state <span class="citation" data-cites="singla2001Molecular hoffman2011Conformational gaertner2004Comparative">(<a href="#ref-singla2001Molecular" role="doc-biblioref">Singla et al. 2001</a>; <a href="#ref-hoffman2011Conformational" role="doc-biblioref">L. Hoffman et al. 2011</a>; <a href="#ref-gaertner2004Comparative" role="doc-biblioref">Gaertner et al. 2004</a>)</span>.</li>
<li>T286 phosphorylation: Phosphorylation at this site occurs at the hinge region of CaMKII and prevents subunits from closing <span class="citation" data-cites="braun1995Multifunctional miller1986Regulation coultrap2010CaMKIIa buard2010CaMKII chang2017CaMKII">(<a href="#ref-braun1995Multifunctional" role="doc-biblioref">Braun and Schulman 1995</a>; <a href="#ref-miller1986Regulation" role="doc-biblioref">Miller and Kennedy 1986</a>; <a href="#ref-coultrap2010CaMKIIa" role="doc-biblioref">Coultrap et al. 2010</a>; <a href="#ref-buard2010CaMKII" role="doc-biblioref">Buard et al. 2010</a>; <a href="#ref-chang2017CaMKII" role="doc-biblioref">Chang et al. 2017</a>)</span>. This modification thus sustains the subunit in an open conformation, contributing to the prolonged activation of the kinase.</li>
<li>NMDAR binding: Once a CaMKII subunit binds to the NMDA receptor, the interaction can stabilise the open state of the kinase. This CaMKII/NMDAR association has been shown to maintain the kinase in an active, open conformation, as discussed in <a href="#sec-CaMKII-NMDAR-association" class="quarto-xref"><span class="quarto-unresolved-ref">sec-CaMKII-NMDAR-association</span></a>.</li>
</ul>
</section>
<section id="sec-binding-cam-camkii" class="level3" data-number="5.2.6">
<h3 data-number="5.2.6" class="anchored" data-anchor-id="sec-binding-cam-camkii"><span class="header-section-number">5.2.6</span> Binding of CaM to CaMKII</h3>
<p>CaM bound to different amounts of calcium has been shown to bind to CaMKII with different affinities, with CaM bound to four calcium having the highest affinity <span class="citation" data-cites="linkevicius2025Fitting ordyan2020Interactions khan2024Realtimea">(<a href="#ref-linkevicius2025Fitting" role="doc-biblioref">Linkevicius et al. 2025</a>; <a href="#ref-ordyan2020Interactions" role="doc-biblioref">Ordyan et al. 2020</a>; <a href="#ref-khan2024Realtimea" role="doc-biblioref">Khan et al. 2024</a>)</span>. While CaM binding to the low-affinity site is compatible with CaMKII remaining in a closed state, binding to the high-affinity site is thought to structurally prevent CaMKII from closing <span class="citation" data-cites="singla2001Molecular hoffman2011Conformational gaertner2004Comparative">(<a href="#ref-singla2001Molecular" role="doc-biblioref">Singla et al. 2001</a>; <a href="#ref-hoffman2011Conformational" role="doc-biblioref">L. Hoffman et al. 2011</a>; <a href="#ref-gaertner2004Comparative" role="doc-biblioref">Gaertner et al. 2004</a>)</span>. The literature suggestes that CaM binding in itself is not sufficient for CaMKII activation, although high-affinity binding of CaM is <span class="citation" data-cites="evans2009Energetics stefan2012Structural bossuyt2013Visualizing">(<a href="#ref-evans2009Energetics" role="doc-biblioref">Evans and Shea 2009</a>; <a href="#ref-stefan2012Structural" role="doc-biblioref">M. I. Stefan, Marshall, and Novère 2012</a>; <a href="#ref-bossuyt2013Visualizing" role="doc-biblioref">Bossuyt and Bers 2013</a>)</span>. This is likely due to the stabilising effect of high affinity binding of CaM to CaMKII structurally keeping the kinase in its active state.</p>
<p>In this model, for simplicity, we therefore assume that only fully saturated CaM with four calcium ions (CaM_Ca4) binds to open CaMKII subunits. Additionally, CaM binds to a CaMKII subunit only if T306 is dephosphorylated. The T305/T306 site lies within the CaM binding region. It has been shown that autophosphorylation at T305/T306 (T306 hereforth) impedes CaM binding, and similarly CaM binding prevents T306 phosphorylation <span class="citation" data-cites="rellos2010Structure">(<a href="#ref-rellos2010Structure" role="doc-biblioref">Rellos, Pike, Niesen, Salah, Lee, von Delft, et al. 2010</a>)</span>. In the model here, CaM is able to bind to CaMKII with the following rules:</p>
<ul>
<li>CaM must be fully saturated (bound to four calcium ions), i.e.&nbsp;only CaM_Ca4 binds to CaMKII. We assume only one high affinity binding site for CaM in each CaMKII subunit.</li>
<li>CaMKII subunit must be in its open/active state.</li>
<li>T306 must be unphosphorylated.</li>
</ul>
<p>As described above in <a href="#sec-camkii-flicker" class="quarto-xref"><span class="quarto-unresolved-ref">sec-camkii-flicker</span></a> when a CaMKII subunit spontaneously flickers into its active/open state CaM binding is likely to stabilize the subunit once it is already active, but does not initiate activation. Similarly, while it is commonly suggested that simultaneous binding of CaM to two neighboring subunits triggers T286 autophosphorylation, we propose that CaM’s role might be more nuanced. Instead of being essential for T286 phosphorylation, CaM may stabilize the open state of CaMKII, thereby increasing the likelihood of phosphorylation. Therefore, in this model, we do not make it a requirement for CaM to be bound to neighbouring subunits for phosphorylation to happen. We consider CaM binding is not strictly necessary for autophosphorylation but rather enhances the probability of phosphorylation by maintaining subunits in their active/open state.</p>
</section>
<section id="camkii-phosphorylation-and-dephosphorylation" class="level3" data-number="5.2.7">
<h3 data-number="5.2.7" class="anchored" data-anchor-id="camkii-phosphorylation-and-dephosphorylation"><span class="header-section-number">5.2.7</span> CaMKII phosphorylation and dephosphorylation</h3>
<p>The most studied phosphorylation sites on CaMKII are T286 and T305/T306. These sites are key for regulating CaMKII activity and subcellular localization at the PSD during LTP, which is of relevance for synaptic plasticity as well as learning and memory <a href="#sec-cam-binding-and-t286-p" class="quarto-xref">see&nbsp;<span class="quarto-unresolved-ref ref-noprefix">sec-cam-binding-and-t286-p</span></a>. Likewise, in this study, we focus on these two sites. While other phosphorylation sites on CaMKII have been identified <span class="citation" data-cites="migues2006Phosphorylation gangopadhyay2008Regulation">(<a href="#ref-migues2006Phosphorylation" role="doc-biblioref">Migues et al. 2006</a>; <a href="#ref-gangopadhyay2008Regulation" role="doc-biblioref">Gangopadhyay et al. 2008</a>)</span>, their functions are not yet fully understood and are not within the scope of this study.</p>
</section>
<section id="t286-phosphorylation" class="level3" data-number="5.2.8">
<h3 data-number="5.2.8" class="anchored" data-anchor-id="t286-phosphorylation"><span class="header-section-number">5.2.8</span> T286 phosphorylation</h3>
<p>I have modelled phosphorylation of T286 to occur with the condition that two adjascent CaMKII subunits are in their open states. Research shows that CaMKII subunits need to be open to expose the ATP binding site in the regulatory domain, allowing the gamma-phosphate of ATP to transfer to the threonine residue at T286 <span class="citation" data-cites="lou1986Activation hoffman2011Investigation">(<a href="#ref-lou1986Activation" role="doc-biblioref">Lou, Lloyd, and Schulman 1986</a>; <a href="#ref-hoffman2011Investigation" role="doc-biblioref">L. R. Hoffman 2011</a>)</span>. The open (kinase domain available) and undocked (subunits can reach their neighbours) states of two neighbouring subunits is crucial for the catalytic site of one subunit to reach and phosphorylate its neighboring subunit. In our model, we assume ATP molecule levels are sufficient to ensure the reaction proceeds at its maximal rate (saturating), so we do not model ATP as a reactant but rather focus on the phosphorylation process under these conditions.</p>
<p>As discussed earlier, CaM bound to two neighbouring subunits is often referred to in the literature as being required for autophosphorylation and subsequent CaMKII activation <span class="citation" data-cites="nicoll2005Synaptic rellos2010Structureb">(<a href="#ref-nicoll2005Synaptic" role="doc-biblioref">Nicoll and Schmitz 2005</a>; <a href="#ref-rellos2010Structureb" role="doc-biblioref">Rellos, Pike, Niesen, Salah, Lee, Delft, et al. 2010</a>)</span>. CaM has been shown to keep CaMKII subunits in their open state, and therefore it may indirectly enhance probability of T286 phosphorylation.</p>
<p>In the model I use here I don’t make CaM a requirement for T286 phosphorylation, but rather once CaMKII is bound to CaM_Ca4, this binding keeps CaMKII subunits in their open state and further enables phosphorylation. The only requirement for CaMKII T286 phosphorylation is, therefore, that two adjacent subunits are in their open state to enable inter subunit phosphorylation at this site (as per <a href="#fig-docked-undocked" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-docked-undocked</span></a> previously described in <a href="#sec-biology-chapter" class="quarto-xref"><span class="quarto-unresolved-ref">sec-biology-chapter</span></a> and below in <a href="#fig-camkii-binding-rxns" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-camkii-binding-rxns</span></a>).</p>
</section>
<section id="t306-phosphorylation" class="level3" data-number="5.2.9">
<h3 data-number="5.2.9" class="anchored" data-anchor-id="t306-phosphorylation"><span class="header-section-number">5.2.9</span> T306 phosphorylation</h3>
<p>In contrast to T286 phosphorylation, T306 phosphorylation prevents binding of CaM, as the phosphate group added to the threonine creates electrostatic and steric interference, occupying the position where CaM normally interacts. In the model used here, CaM can therefore only bind to CaMKII if the T306 site is not phosphorylated, and conversely, phosphorylation at T306 cannot occur while CaM is bound (previously discussed in <a href="#sec-camkii-as-dodecamer" class="quarto-xref"><span class="quarto-unresolved-ref">sec-camkii-as-dodecamer</span></a>). Similar to the conditions for T286, the model requires that a subunit must be in the open state for phosphorylation at T306 to take place. Since T286 phosphorylation maintains subunits in their open state, phosphorylation at T306 is more likely to occur if a neighboring subunit is phosphorylated at T286.</p>
<p>An aspect of phosphorylation at this site is that it does not prevent CaMKII subunits from closing, unlike T286 phosphorylation <span class="citation" data-cites="cook2021CaMKIIb">(<a href="#ref-cook2021CaMKIIb" role="doc-biblioref">Cook et al. 2021</a>)</span>. This, together with the fact that it prevents CaM from binding, makes it more likely that subunits that are phosphorylated at T306 are found in the closed state too. When the subunit is closed, catalytic activity of the kinase is reduced, as the kinase domain is not accessible to perform catalytic activity. Therefore, this phosphorylation is ofen refered to as reducing catalytic activity of CaMKII. T306 phosphorylation may act as a form of kinetic proofreading, by ensuring that CaMKII remains inactive, or less active, unless sustained or repetitive calcium signals are present. Phosphorylation at T306 likely prevents premature or excessive kinase activity, thereby enhancing the specificity and fidelity of CaMKII-mediated signalling in response to synaptic activity, which is discussed more in depth in [INSERT SECTION REFERENCE].</p>
</section>
<section id="dephosphorylation-of-camkii-subunits" class="level3" data-number="5.2.10">
<h3 data-number="5.2.10" class="anchored" data-anchor-id="dephosphorylation-of-camkii-subunits"><span class="header-section-number">5.2.10</span> Dephosphorylation of CaMKII subunits</h3>
<p>There are multiple known phosphatases which act to dephosphorylate CaMKII in the PSD <span class="citation" data-cites="cohen1997Novel colbran2004Protein">(<a href="#ref-cohen1997Novel" role="doc-biblioref">Cohen 1997</a>; <a href="#ref-colbran2004Protein" role="doc-biblioref">Colbran 2004</a>)</span>. Different phosphatase proteins will carry out their function depending on their localization and what protein assemblies they are bound to or are available for them. The Ser/Thr phosphatase Protein Phosphatase-1 (PP1) is a key regulator of CaMKII signalling <span class="citation" data-cites="shioda2017Physiological">(<a href="#ref-shioda2017Physiological" role="doc-biblioref">Shioda and Fukunaga 2017</a>)</span>; and it can dephosphorylate both T286 and T306 sites of the kinase.</p>
<p>The colocalization with PSD-associated PP1 or sequestration away from other protein phosphatases, such as protein phosphatase 2A or protein phosphatase 2B (PP2A, PP2B), results in differential modulation of CaMKII’s T286 dephosphorylation. PP1 can form a complex with over 50 regulatory or scaffolding proteins that dictate substrate specificity and subcellular localization. There are four known PP1 subunit isoforms of the catalytic subunit (<span class="math inline">\(\alpha\)</span>, <span class="math inline">\(\beta\)</span>, <span class="math inline">\(\gamma\)</span>-1, and <span class="math inline">\(\gamma\)</span>-2), each interacting with over 40 known regulatory and target subunits. PP1-<span class="math inline">\(\alpha\)</span> has been shown to account for the majority of phosphatase activity with CaMKII in the PSD <span class="citation" data-cites="colbran2004Protein">(<a href="#ref-colbran2004Protein" role="doc-biblioref">Colbran 2004</a>)</span>. Therefore, we focus on PP1-<span class="math inline">\(\alpha\)</span> in this study, and refer to it simply as PP1.</p>
<p>In the model used here, PP1 catalyzes the dephosphorylation of CaMKII without forming a stable enzyme-substrate complex. Instead, the reaction proceeds in a manner where PP1 transiently interacts, without binding, with a CaMKII subunit under specific conditions, facilitating the dephosphorylation process. PP1 can only dephosphorylate CaMKII subunits when they are unbound from CaM, as CaM structurally obstructs access to the T286 phosphorylation site. Additionally, the CaMKII subunit must be in its open conformation to expose both the T286 and T306 sites for PP1-mediated dephosphorylation.</p>
</section>
<section id="reaction-rules-and-kinetic-rates" class="level3" data-number="5.2.11">
<h3 data-number="5.2.11" class="anchored" data-anchor-id="reaction-rules-and-kinetic-rates"><span class="header-section-number">5.2.11</span> Reaction rules and kinetic rates</h3>
<p>In BNGL, reaction rules are specified as discrete transformations. For example, the rule “CaM(ca~0,camkii) + Ca() &lt;-&gt; CaM(ca~1,camkii)” represents the binding of a single calcium ion to CaM that is not bound to CaMKII (see reaction #1 in <a href="#fig-tbl-1" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-tbl-1</span></a>). The assigned rate constants for each rule are used to calculate the probability of the reaction occurring when the two involved molecules collide, or at each time step for unimolecular transformations.</p>
<p>As molecules diffuse within the simulation volume, all molecules within a specified radius along a given trajectory or at the point of collision on a surface are considered for potential reactions. For example, when a calcium ion molecule encounters a free CaM molecule at the point of collision, they react according to the specified rule in the model, resulting in the formation of the molecule CaM_Ca1. CaM_Ca1 can continue to diffuse based on the parameters defined in the simulation and can interact with other molecules as well (see back for diagram example explanation in <a href="#fig-mcell-works" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-mcell-works</span></a>).</p>
<p>The reaction rules defined below follow the “don’t care, don’t write” principle of rule-based modelling: if a molecular feature is irrelevant to a particular reaction, it is simply omitted from the reaction definition. For example, in reaction #5, which describes the flickering of CaMKII subunits between open and closed states, the conditions specified are that the subunit’s CaM-binding site must be free, the subunit must be unphosphorylated at T286, and the subunit must be open. However, the phosphorylation state at T306 is irrelevant to this reaction, and therefore is not specified.</p>
<div id="fig-tbl-1" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="Reactions and rate constants used in the model" data-fig-pos="'H">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-tbl-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="30-methods-figures\table_rxns1.PNG" class="img-fluid figure-img" data-fig-pos="'H" data-fig-scap="Reactions and rate constants used in the model">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-tbl-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5.5: Reaction rules and rate constants used in the model. <span class="math inline">\(k_\text{on}\)</span> (association rates) in M<span class="math inline">\(^{-1}\)</span>s<span class="math inline">\(^{-1}\)</span>, <span class="math inline">\(k_\text{off}\)</span> (dissociation rates) in s<span class="math inline">\(^{-1}\)</span>, <span class="math inline">\(k_\text{cat}\)</span> (catalytic rates) in s<span class="math inline">\(^{-1}\)</span>.
</figcaption>
</figure>
</div>
<div id="fig-camkii-binding-rxns" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-pos="H">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-camkii-binding-rxns-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="30-methods-figures\camkii-binding-rxns.PNG" class="img-fluid figure-img" style="width:90.0%" data-fig-pos="H">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-camkii-binding-rxns-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5.6: Reaction rules. #d is missing. all possible iterations of a camkii subunit
</figcaption>
</figure>
</div>


<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-akashi2009NMDA" class="csl-entry" role="listitem">
Akashi, Kaori, Toshikazu Kakizaki, Haruyuki Kamiya, Masahiro Fukaya, Miwako Yamasaki, Manabu Abe, Rie Natsume, Masahiko Watanabe, and Kenji Sakimura. 2009. <span>“<span>NMDA Receptor GluN2B</span> (<span>GluRε2</span>/<span>NR2B</span>) <span>Subunit Is Crucial</span> for <span>Channel Function</span>, <span>Postsynaptic Macromolecular Organization</span>, and <span>Actin Cytoskeleton</span> at <span>Hippocampal CA3 Synapses</span>.”</span> <em>J. Neurosci.</em> 29 (35): 10869–82. <a href="https://doi.org/10.1523/JNEUROSCI.5531-08.2009">https://doi.org/10.1523/JNEUROSCI.5531-08.2009</a>.
</div>
<div id="ref-asciutto2021Physicalc" class="csl-entry" role="listitem">
Asciutto, Eliana K., Sergio Pantano, and Ignacio J. General. 2021. <span>“Physical Interactions Driving the Activation/Inhibition of Calcium/Calmodulin Dependent Protein Kinase <span>II</span>.”</span> <em>Journal of Molecular Graphics and Modelling</em> 105 (June): 107875. <a href="https://doi.org/10.1016/j.jmgm.2021.107875">https://doi.org/10.1016/j.jmgm.2021.107875</a>.
</div>
<div id="ref-barcomb2016CaMKIIa" class="csl-entry" role="listitem">
Barcomb, Kelsey, Johannes W. Hell, Tim A. Benke, and K. Ulrich Bayer. 2016. <span>“The <span>CaMKII</span>/<span>GluN2B Protein Interaction Maintains Synaptic Strength</span> *.”</span> <em>Journal of Biological Chemistry</em> 291 (31): 16082–89. <a href="https://doi.org/10.1074/jbc.M116.734822">https://doi.org/10.1074/jbc.M116.734822</a>.
</div>
<div id="ref-bartol2024Spatial" class="csl-entry" role="listitem">
Bartol, Thomas M., Mariam Ordyan, Terrence J. Sejnowski, Padmini Rangamani, and Mary B. Kennedy. 2024. <span>“A Spatial Model of Autophosphorylation of <span>CaMKII</span> in a Glutamatergic Spine Suggests a Network-Driven Kinetic Mechanism for Bistable Changes in Synaptic Strength.”</span> February 2, 2024. <a href="https://doi.org/10.1101/2024.02.02.578696">https://doi.org/10.1101/2024.02.02.578696</a>.
</div>
<div id="ref-bayer2006Transitionb" class="csl-entry" role="listitem">
Bayer, K. Ulrich, Eric LeBel, Greg L. McDonald, Heather O’Leary, Howard Schulman, and Paul De Koninck. 2006. <span>“Transition from Reversible to Persistent Binding of <span>CaMKII</span> to Postsynaptic Sites and <span>NR2B</span>.”</span> <em>J Neurosci</em> 26 (4): 1164–74. <a href="https://doi.org/10.1523/JNEUROSCI.3116-05.2006">https://doi.org/10.1523/JNEUROSCI.3116-05.2006</a>.
</div>
<div id="ref-bayer2019CaM" class="csl-entry" role="listitem">
Bayer, K. Ulrich, and Howard Schulman. 2019. <span>“<span>CaM Kinase</span>: <span>Still Inspiring</span> at 40.”</span> <em>Neuron</em> 103 (3): 380–94. <a href="https://doi.org/10.1016/j.neuron.2019.05.033">https://doi.org/10.1016/j.neuron.2019.05.033</a>.
</div>
<div id="ref-bayer2001Interaction" class="csl-entry" role="listitem">
Bayer, K.-Ulrich, Paul De Koninck, A. Soren Leonard, Johannes W. Hell, and Howard Schulman. 2001. <span>“Interaction with the <span>NMDA</span> Receptor Locks <span>CaMKII</span> in an Active Conformation.”</span> <em>Nature</em> 411 (6839): 801–5. <a href="https://doi.org/10.1038/35081080">https://doi.org/10.1038/35081080</a>.
</div>
<div id="ref-bossuyt2013Visualizing" class="csl-entry" role="listitem">
Bossuyt, Julie, and Donald M. Bers. 2013. <span>“Visualizing <span>CaMKII</span> and <span>CaM</span> Activity: A Paradigm of Compartmentalized Signaling.”</span> <em>J Mol Med</em> 91 (8): 907–16. <a href="https://doi.org/10.1007/s00109-013-1060-y">https://doi.org/10.1007/s00109-013-1060-y</a>.
</div>
<div id="ref-braun1995Multifunctional" class="csl-entry" role="listitem">
Braun, A. P., and H. Schulman. 1995. <span>“The Multifunctional Calcium/Calmodulin-Dependent Protein Kinase: From Form to Function.”</span> <em>Annu Rev Physiol</em> 57: 417–45. <a href="https://doi.org/10.1146/annurev.ph.57.030195.002221">https://doi.org/10.1146/annurev.ph.57.030195.002221</a>.
</div>
<div id="ref-buard2010CaMKII" class="csl-entry" role="listitem">
Buard, Isabelle, Steven J. Coultrap, Ronald K. Freund, Yong-Seok Lee, Mark L. Dell’Acqua, Alcino J. Silva, and K. Ulrich Bayer. 2010. <span>“<span>CaMKII</span> <span>‘<span>Autonomy</span>’</span> <span>Is Required</span> for <span>Initiating But Not</span> for <span>Maintaining Neuronal Long-Term Information Storage</span>.”</span> <em>J. Neurosci.</em> 30 (24): 8214–20. <a href="https://doi.org/10.1523/JNEUROSCI.1469-10.2010">https://doi.org/10.1523/JNEUROSCI.1469-10.2010</a>.
</div>
<div id="ref-chang2017CaMKII" class="csl-entry" role="listitem">
Chang, Jui-Yun, Paula Parra-Bueno, Tal Laviv, Erzsebet M. Szatmari, Seok-Jin R. Lee, and Ryohei Yasuda. 2017. <span>“<span>CaMKII Autophosphorylation Is Necessary</span> for <span>Optimal Integration</span> of <span>Ca2</span>+ <span>Signals</span> During <span>LTP Induction</span>, but <span>Not Maintenance</span>.”</span> <em>Neuron</em> 94 (4): 800–808.e4. <a href="https://doi.org/10.1016/j.neuron.2017.04.041">https://doi.org/10.1016/j.neuron.2017.04.041</a>.
</div>
<div id="ref-cohen1997Novel" class="csl-entry" role="listitem">
Cohen, Patricia T. W. 1997. <span>“Novel Protein Serine/Threonine Phosphatases: <span>Variety</span> Is the Spice of Life.”</span> <em>Trends in Biochemical Sciences</em> 22 (7): 245–51. <a href="https://doi.org/10.1016/S0968-0004(97)01060-8">https://doi.org/10.1016/S0968-0004(97)01060-8</a>.
</div>
<div id="ref-colbran2004Protein" class="csl-entry" role="listitem">
Colbran, Roger J. 2004. <span>“Protein <span>Phosphatases</span> and <span>Calcium</span>/<span>Calmodulin-Dependent Protein Kinase II-Dependent Synaptic Plasticity</span>.”</span> <em>J. Neurosci.</em> 24 (39): 8404–9. <a href="https://doi.org/10.1523/JNEUROSCI.3602-04.2004">https://doi.org/10.1523/JNEUROSCI.3602-04.2004</a>.
</div>
<div id="ref-cook2021CaMKIIb" class="csl-entry" role="listitem">
Cook, S. G., O. R. Buonarati, S. J. Coultrap, and K. U. Bayer. 2021. <span>“<span>CaMKII</span> Holoenzyme Mechanisms That Govern the <span>LTP</span> Versus <span>LTD</span> Decision.”</span> <em>Science Advances</em> 7 (16). <a href="https://doi.org/10.1126/sciadv.abe2300">https://doi.org/10.1126/sciadv.abe2300</a>.
</div>
<div id="ref-coultrap2010CaMKIIa" class="csl-entry" role="listitem">
Coultrap, Steven J., Isabelle Buard, Jaqueline R. Kulbe, Mark L. Dell’Acqua, and K. Ulrich Bayer. 2010. <span>“<span class="nocase">CaMKII Autonomy Is Substrate-dependent</span> and <span>Further Stimulated</span> by <span>Ca2</span>+/<span>Calmodulin</span>*.”</span> <em>Journal of Biological Chemistry</em> 285 (23): 17930–37. <a href="https://doi.org/10.1074/jbc.M109.069351">https://doi.org/10.1074/jbc.M109.069351</a>.
</div>
<div id="ref-davies2007ACaMKII" class="csl-entry" role="listitem">
Davies, Kurtis D., Rachel M. Alvestad, Steven J. Coultrap, and Michael D. Browning. 2007. <span>“<span class="nocase">αCaMKII</span> Autophosphorylation Levels Differ Depending on Subcellular Localization.”</span> <em>Brain Research</em> 1158 (July): 39–49. <a href="https://doi.org/10.1016/j.brainres.2007.05.008">https://doi.org/10.1016/j.brainres.2007.05.008</a>.
</div>
<div id="ref-evans2009Energetics" class="csl-entry" role="listitem">
Evans, T. Idil Apak, and Madeline A. Shea. 2009. <span>“Energetics of Calmodulin Domain Interactions with the Calmodulin Binding Domain of <span>CaMKII</span>.”</span> <em>Proteins: Structure, Function, and Bioinformatics</em> 76 (1): 47–61. <a href="https://doi.org/10.1002/prot.22317">https://doi.org/10.1002/prot.22317</a>.
</div>
<div id="ref-gaertner2004Comparative" class="csl-entry" role="listitem">
Gaertner, Tara R., Steven J. Kolodziej, Dan Wang, Ryuji Kobayashi, John M. Koomen, James K. Stoops, and M. Neal Waxham. 2004. <span>“Comparative <span>Analyses</span> of the <span class="nocase">Three-dimensional Structures</span> and <span>Enzymatic Properties</span> of α, β, γ, and δ <span>Isoforms</span> of <span>Ca2</span>+-<span class="nocase">Calmodulin-dependent Protein Kinase II</span> *.”</span> <em>Journal of Biological Chemistry</em> 279 (13): 12484–94. <a href="https://doi.org/10.1074/jbc.M313597200">https://doi.org/10.1074/jbc.M313597200</a>.
</div>
<div id="ref-gangopadhyay2008Regulation" class="csl-entry" role="listitem">
Gangopadhyay, Samudra S., Cynthia Gallant, Eric J. Sundberg, William S. Lane, and Kathleen G. Morgan. 2008. <span>“Regulation of <span>Ca2</span>+/Calmodulin Kinase <span>II</span> by a Small <span class="nocase">C-terminal</span> Domain Phosphatase.”</span> <em>Biochemical Journal</em> 412 (3): 507–16. <a href="https://doi.org/10.1042/BJ20071582">https://doi.org/10.1042/BJ20071582</a>.
</div>
<div id="ref-gonzalez-andrade2016Insights" class="csl-entry" role="listitem">
González-Andrade, Martin, Rodríguez-Sotres, Madariaga-Mazón, Rivera-Chávez, Mata, Sosa-Peinado, Pozo-Yauner, and Imilla I. and Arias-Olguín. 2016. <span>“Insights into Molecular Interactions Between <span>CaM</span> and Its Inhibitors from Molecular Dynamics Simulations and Experimental Data.”</span> <em>Journal of Biomolecular Structure and Dynamics</em> 34 (1): 78–91. <a href="https://doi.org/10.1080/07391102.2015.1022225">https://doi.org/10.1080/07391102.2015.1022225</a>.
</div>
<div id="ref-goodell2014CaMKII" class="csl-entry" role="listitem">
Goodell, Dayton J., Tatiana A. Eliseeva, Steven J. Coultrap, and K. Ulrich Bayer. 2014. <span>“<span>CaMKII Binding</span> to <span>GluN2B Is Differentially Affected</span> by <span>Macromolecular Crowding Reagents</span>.”</span> <em>PLOS ONE</em> 9 (5): e96522. <a href="https://doi.org/10.1371/journal.pone.0096522">https://doi.org/10.1371/journal.pone.0096522</a>.
</div>
<div id="ref-hoffman2011Investigation" class="csl-entry" role="listitem">
Hoffman, Laurel R. 2011. <span>“Investigation of <span>CaMKII</span> Activation: A Model of Self-Regulation,”</span> December. <a href="https://etd.library.vanderbilt.edu/etd-11302011-144712">https://etd.library.vanderbilt.edu/etd-11302011-144712</a>.
</div>
<div id="ref-hoffman2011Conformational" class="csl-entry" role="listitem">
Hoffman, Laurel, Richard A. Stein, Roger J. Colbran, and Hassane S. Mchaourab. 2011. <span>“Conformational Changes Underlying Calcium/Calmodulin-Dependent Protein Kinase <span>II</span> Activation.”</span> <em>EMBO J</em> 30 (7): 1251–62. <a href="https://doi.org/10.1038/emboj.2011.40">https://doi.org/10.1038/emboj.2011.40</a>.
</div>
<div id="ref-holmes2000Models" class="csl-entry" role="listitem">
Holmes, William R. 2000. <span>“Models of <span>Calmodulin Trapping</span> and <span>CaM Kinase II Activation</span> in a <span>Dendritic Spine</span>.”</span> <em>J Comput Neurosci</em> 8 (1): 65–86. <a href="https://doi.org/10.1023/A:1008969032563">https://doi.org/10.1023/A:1008969032563</a>.
</div>
<div id="ref-johnson2015Model" class="csl-entry" role="listitem">
Johnson, Todd, Tom Bartol, Terrence Sejnowski, and Eric Mjolsness. 2015. <span>“Model Reduction for Stochastic <span>CaMKII</span> Reaction Kinetics in Synapses by Graph-Constrained Correlation Dynamics.”</span> <em>Phys. Biol.</em> 12 (4): 045005. <a href="https://doi.org/10.1088/1478-3975/12/4/045005">https://doi.org/10.1088/1478-3975/12/4/045005</a>.
</div>
<div id="ref-kawasaki2017Conformational" class="csl-entry" role="listitem">
Kawasaki, Hiroshi, and Robert H. Kretsinger. 2017. <span>“Conformational Landscape Mapping the Difference Between <span class="nocase">N-lobes</span> and <span class="nocase">C-lobes</span> of Calmodulin.”</span> <em>Journal of Inorganic Biochemistry</em> 177 (December): 55–62. <a href="https://doi.org/10.1016/j.jinorgbio.2017.08.025">https://doi.org/10.1016/j.jinorgbio.2017.08.025</a>.
</div>
<div id="ref-khan2024Realtimea" class="csl-entry" role="listitem">
Khan, Shahid, Justin E. Molloy, Henry Puhl, Howard Schulman, and Steven S. Vogel. 2024. <span>“Real-Time Single-Molecule Imaging of <span class="nocase">CaMKII-calmodulin</span> Interactions.”</span> <em>Biophysical Journal</em> 123 (7): 824–38. <a href="https://doi.org/10.1016/j.bpj.2024.02.021">https://doi.org/10.1016/j.bpj.2024.02.021</a>.
</div>
<div id="ref-lenovere2001StochSim" class="csl-entry" role="listitem">
Le Novère, Nicolas, and Thomas Simon Shimizu. 2001. <span>“<span>StochSim</span>: Modelling of Stochastic Biomolecular Processes.”</span> <em>Bioinformatics</em> 17 (6): 575–76. <a href="https://doi.org/10.1093/bioinformatics/17.6.575">https://doi.org/10.1093/bioinformatics/17.6.575</a>.
</div>
<div id="ref-linkevicius2025Fitting" class="csl-entry" role="listitem">
Linkevicius, Domas, Angus Chadwick, Guido C. Faas, Melanie I. Stefan, and David C. Sterratt. 2025. <span>“Fitting and Comparison of Calcium-Calmodulin Kinetic Schemes to a Common Data Set Using Non-Linear Mixed Effects Modelling.”</span> <em>PLoS One</em> 20 (2): e0318646. <a href="https://doi.org/10.1371/journal.pone.0318646">https://doi.org/10.1371/journal.pone.0318646</a>.
</div>
<div id="ref-lisman2001Model" class="csl-entry" role="listitem">
Lisman, John E., and Anatol M. Zhabotinsky. 2001. <span>“A <span>Model</span> of <span>Synaptic Memory</span>: <span>A CaMKII</span>/<span>PP1 Switch</span> That <span>Potentiates Transmission</span> by <span>Organizing</span> an <span>AMPA Receptor Anchoring Assembly</span>.”</span> <em>Neuron</em> 31 (2): 191–201. <a href="https://doi.org/10.1016/S0896-6273(01)00364-6">https://doi.org/10.1016/S0896-6273(01)00364-6</a>.
</div>
<div id="ref-lisman2002Molecular" class="csl-entry" role="listitem">
Lisman, J., H. Schulman, and H. Cline. 2002. <span>“The Molecular Basis of <span>CaMKII</span> Function in Synaptic and Behavioural Memory.”</span> <em>Nat Rev Neurosci</em> 3 (3): 175–90. <a href="https://doi.org/10.1038/nrn753">https://doi.org/10.1038/nrn753</a>.
</div>
<div id="ref-lou1986Activation" class="csl-entry" role="listitem">
Lou, L L, S J Lloyd, and H Schulman. 1986. <span>“Activation of the Multifunctional <span>Ca2</span>+/Calmodulin-Dependent Protein Kinase by Autophosphorylation: <span>ATP</span> Modulates Production of an Autonomous Enzyme.”</span> <em>Proceedings of the National Academy of Sciences</em> 83 (24): 9497–9501. <a href="https://doi.org/10.1073/pnas.83.24.9497">https://doi.org/10.1073/pnas.83.24.9497</a>.
</div>
<div id="ref-lucic2008Detaileda" class="csl-entry" role="listitem">
Lučić, Vladan, Gabriela J. Greif, and Mary B. Kennedy. 2008. <span>“Detailed State Model of <span>CaMKII</span> Activation and Autophosphorylation.”</span> <em>Eur Biophys J</em> 38 (1): 83–98. <a href="https://doi.org/10.1007/s00249-008-0362-4">https://doi.org/10.1007/s00249-008-0362-4</a>.
</div>
<div id="ref-migues2006Phosphorylation" class="csl-entry" role="listitem">
Migues, Paola V., Ingo T. Lehmann, Lisa Fluechter, Martin Cammarota, James W. Gurd, Alistair T. R. Sim, Phillip W. Dickson, and John A. P. Rostas. 2006. <span>“Phosphorylation of <span>CaMKII</span> at <span>Thr253</span> Occurs in Vivo and Enhances Binding to Isolated Postsynaptic Densities.”</span> <em>Journal of Neurochemistry</em> 98 (1): 289–99. <a href="https://doi.org/10.1111/j.1471-4159.2006.03876.x">https://doi.org/10.1111/j.1471-4159.2006.03876.x</a>.
</div>
<div id="ref-miller1986Regulation" class="csl-entry" role="listitem">
Miller, S. G., and M. B. Kennedy. 1986. <span>“Regulation of Brain Type <span>II Ca2</span>+/Calmodulin-Dependent Protein Kinase by Autophosphorylation: A <span>Ca2</span>+-Triggered Molecular Switch.”</span> <em>Cell</em> 44 (6): 861–70. <a href="https://doi.org/10.1016/0092-8674(86)90008-5">https://doi.org/10.1016/0092-8674(86)90008-5</a>.
</div>
<div id="ref-myers2017CaMKII" class="csl-entry" role="listitem">
Myers, Janette B., Vincent Zaegel, Steven J. Coultrap, Adam P. Miller, K. Ulrich Bayer, and Steve L. Reichow. 2017. <span>“The <span>CaMKII</span> Holoenzyme Structure in Activation-Competent Conformations.”</span> <em>Nat Commun</em> 8 (June): 15742. <a href="https://doi.org/10.1038/ncomms15742">https://doi.org/10.1038/ncomms15742</a>.
</div>
<div id="ref-nicoll2005Synaptic" class="csl-entry" role="listitem">
Nicoll, Roger A., and Dietmar Schmitz. 2005. <span>“Synaptic Plasticity at Hippocampal Mossy Fibre Synapses.”</span> <em>Nat Rev Neurosci</em> 6 (11): 863–76. <a href="https://doi.org/10.1038/nrn1786">https://doi.org/10.1038/nrn1786</a>.
</div>
<div id="ref-nicoll2023Synaptic" class="csl-entry" role="listitem">
Nicoll, Roger A., and Howard Schulman. 2023. <span>“Synaptic Memory and <span>CaMKII</span>.”</span> <em>Physiol Rev</em> 103 (4): 2897–2945. <a href="https://doi.org/10.1152/physrev.00034.2022">https://doi.org/10.1152/physrev.00034.2022</a>.
</div>
<div id="ref-ordyan2020Interactions" class="csl-entry" role="listitem">
Ordyan, Mariam, Tom Bartol, Mary Kennedy, Padmini Rangamani, and Terrence Sejnowski. 2020. <span>“Interactions Between Calmodulin and Neurogranin Govern the Dynamics of <span>CaMKII</span> as a Leaky Integrator.”</span> <em>PLOS Computational Biology</em> 16 (7): e1008015. <a href="https://doi.org/10.1371/journal.pcbi.1008015">https://doi.org/10.1371/journal.pcbi.1008015</a>.
</div>
<div id="ref-ozden2022CaMKIIb" class="csl-entry" role="listitem">
Özden, Can, Roman Sloutsky, Tomohiro Mitsugi, Nicholas Santos, Emily Agnello, Christl Gaubitz, Joshua Foster, et al. 2022. <span>“<span>CaMKII</span> Binds Both Substrates and Activators at the Active Site.”</span> <em>Cell Reports</em> 40 (2). <a href="https://doi.org/10.1016/j.celrep.2022.111064">https://doi.org/10.1016/j.celrep.2022.111064</a>.
</div>
<div id="ref-pepke2010Dynamic" class="csl-entry" role="listitem">
Pepke, Shirley, Tamara Kinzer-Ursem, Stefan Mihalas, and Mary B. Kennedy. 2010. <span>“A <span>Dynamic Model</span> of <span>Interactions</span> of <span>Ca2</span>+, <span>Calmodulin</span>, and <span>Catalytic Subunits</span> of <span>Ca2</span>+/<span>Calmodulin-Dependent Protein Kinase II</span>.”</span> <em>PLOS Computational Biology</em> 6 (2): e1000675. <a href="https://doi.org/10.1371/journal.pcbi.1000675">https://doi.org/10.1371/journal.pcbi.1000675</a>.
</div>
<div id="ref-pharris2019Multistate" class="csl-entry" role="listitem">
Pharris, Matthew C., Neal M. Patel, Tyler G. VanDyk, Thomas M. Bartol, Terrence J. Sejnowski, Mary B. Kennedy, Melanie I. Stefan, and Tamara L. Kinzer-Ursem. 2019. <span>“A Multi-State Model of the <span>CaMKII</span> Dodecamer Suggests a Role for Calmodulin in Maintenance of Autophosphorylation.”</span> <em>PLOS Computational Biology</em> 15 (12): e1006941. <a href="https://doi.org/10.1371/journal.pcbi.1006941">https://doi.org/10.1371/journal.pcbi.1006941</a>.
</div>
<div id="ref-pullara2017Mechanisms" class="csl-entry" role="listitem">
Pullara, Filippo, Eliana K. Asciutto, and Ignacio J. General. 2017. <span>“Mechanisms of <span>Activation</span> and <span>Subunit Release</span> in <span>Ca2</span>+/<span>Calmodulin-Dependent Protein Kinase II</span>.”</span> <em>J. Phys. Chem. B</em> 121 (45): 10344–52. <a href="https://doi.org/10.1021/acs.jpcb.7b09214">https://doi.org/10.1021/acs.jpcb.7b09214</a>.
</div>
<div id="ref-rellos2010Structure" class="csl-entry" role="listitem">
Rellos, Peter, Ashley C. W. Pike, Frank H. Niesen, Eidarus Salah, Wen Hwa Lee, Frank von Delft, and Stefan Knapp. 2010. <span>“Structure of the <span>CaMKIIδ</span>/<span>Calmodulin Complex Reveals</span> the <span>Molecular Mechanism</span> of <span>CaMKII Kinase Activation</span>.”</span> <em>PLoS Biol</em> 8 (7): e1000426. <a href="https://doi.org/10.1371/journal.pbio.1000426">https://doi.org/10.1371/journal.pbio.1000426</a>.
</div>
<div id="ref-rellos2010Structureb" class="csl-entry" role="listitem">
Rellos, Peter, Ashley C. W. Pike, Frank H. Niesen, Eidarus Salah, Wen Hwa Lee, Frank von Delft, and Stefan Knapp. 2010. <span>“Structure of the <span>CaMKIIδ</span>/<span>Calmodulin Complex Reveals</span> the <span>Molecular Mechanism</span> of <span>CaMKII Kinase Activation</span>.”</span> <em>PLOS Biology</em> 8 (7): e1000426. <a href="https://doi.org/10.1371/journal.pbio.1000426">https://doi.org/10.1371/journal.pbio.1000426</a>.
</div>
<div id="ref-rumian2024LTP" class="csl-entry" role="listitem">
Rumian, Nicole L., C. Madison Barker, Matthew E. Larsen, Jonathan E. Tullis, Ronald K. Freund, Amir Taslimi, Steven J. Coultrap, Chandra L. Tucker, Mark L. Dell’Acqua, and K. Ulrich Bayer. 2024. <span>“<span>LTP</span> Expression Mediated by Autonomous Activity of <span class="nocase">GluN2B-bound CaMKII</span>.”</span> <em>Cell Reports</em> 43 (10). <a href="https://doi.org/10.1016/j.celrep.2024.114866">https://doi.org/10.1016/j.celrep.2024.114866</a>.
</div>
<div id="ref-sanabria2008Spatial" class="csl-entry" role="listitem">
Sanabria, Hugo, Michelle A. Digman, Enrico Gratton, and M. Neal Waxham. 2008. <span>“Spatial <span>Diffusivity</span> and <span>Availability</span> of <span>Intracellular Calmodulin</span>.”</span> <em>Biophys J</em> 95 (12): 6002–15. <a href="https://doi.org/10.1529/biophysj.108.138974">https://doi.org/10.1529/biophysj.108.138974</a>.
</div>
<div id="ref-shioda2017Physiological" class="csl-entry" role="listitem">
Shioda, Norifumi, and Kohji Fukunaga. 2017. <span>“Physiological and <span>Pathological Roles</span> of <span>CaMKII-PP1 Signaling</span> in the <span>Brain</span>.”</span> <em>Int J Mol Sci</em> 19 (1): 20. <a href="https://doi.org/10.3390/ijms19010020">https://doi.org/10.3390/ijms19010020</a>.
</div>
<div id="ref-singla2001Molecular" class="csl-entry" role="listitem">
Singla, Sheela I., Andy Hudmon, Jonathan M. Goldberg, Janet L. Smith, and Howard Schulman. 2001. <span>“Molecular <span>Characterization</span> of <span>Calmodulin Trapping</span> by <span>Calcium</span>/<span class="nocase">Calmodulin-dependent Protein Kinase II</span> *.”</span> <em>Journal of Biological Chemistry</em> 276 (31): 29353–60. <a href="https://doi.org/10.1074/jbc.M101744200">https://doi.org/10.1074/jbc.M101744200</a>.
</div>
<div id="ref-stefan2008Allostericd" class="csl-entry" role="listitem">
Stefan, Melanie I., Stuart J. Edelstein, and Nicolas Le Novère. 2008. <span>“An Allosteric Model of Calmodulin Explains Differential Activation of <span>PP2B</span> and <span>CaMKII</span>.”</span> <em>Proceedings of the National Academy of Sciences</em> 105 (31): 10768–73. <a href="https://doi.org/10.1073/pnas.0804672105">https://doi.org/10.1073/pnas.0804672105</a>.
</div>
<div id="ref-stefan2012Structural" class="csl-entry" role="listitem">
Stefan, Melanie I., David P. Marshall, and Nicolas Le Novère. 2012. <span>“Structural <span>Analysis</span> and <span>Stochastic Modelling Suggest</span> a <span>Mechanism</span> for <span>Calmodulin Trapping</span> by <span>CaMKII</span>.”</span> <em>PLOS ONE</em> 7 (1): e29406. <a href="https://doi.org/10.1371/journal.pone.0029406">https://doi.org/10.1371/journal.pone.0029406</a>.
</div>
<div id="ref-stefan2011MultiStage" class="csl-entry" role="listitem">
Stefan, Melanie, Shirley Pepke, Stefan Mihalas, Thomas M. Bartol, Terrence J. Sejnowski, and Mary B. Kennedy. 2011. <span>“Multi-<span>Stage Modeling</span> of the <span>Kinetics</span> of <span>Activation</span> of <span>CaMKII</span>.”</span> In <em>Multi-<span>Stage Modeling</span> of the <span>Kinetics</span> of <span>Activation</span> of <span>CaMKII</span></em>. <a href="https://doi.org/10.3389/conf.fninf.2011.08.00033">https://doi.org/10.3389/conf.fninf.2011.08.00033</a>.
</div>
<div id="ref-zhabotinsky2000Bistability" class="csl-entry" role="listitem">
Zhabotinsky, Anatol M. 2000. <span>“Bistability in the <span>Ca2</span>+/<span>Calmodulin-Dependent Protein Kinase-Phosphatase System</span>.”</span> <em>Biophysical Journal</em> 79 (5): 2211–21. <a href="https://doi.org/10.1016/S0006-3495(00)76469-1">https://doi.org/10.1016/S0006-3495(00)76469-1</a>.
</div>
<div id="ref-zhang2017Opposing" class="csl-entry" role="listitem">
Zhang, Pengzhi, Swarnendu Tripathi, Hoa Trinh, and Margaret S. Cheung. 2017. <span>“Opposing <span>Intermolecular Tuning</span> of <span>Ca2</span>+ <span>Affinity</span> for <span>Calmodulin</span> by <span>Neurogranin</span> and <span>CaMKII Peptides</span>.”</span> <em>Biophysical Journal</em> 112 (6): 1105–19. <a href="https://doi.org/10.1016/j.bpj.2017.01.020">https://doi.org/10.1016/j.bpj.2017.01.020</a>.
</div>
</div>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./20-data-hazards.html" class="pagination-link" aria-label="A Case Study in Ethical Reflection: Data Hazards in this PhD">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">A Case Study in Ethical Reflection: Data Hazards in this PhD</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./40-model-validation.html" class="pagination-link" aria-label="Validating model behaviour">
        <span class="nav-page-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Validating model behaviour</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>