<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.43">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>7&nbsp; Analysis of CaMKII Dynamics in Wild-Type and Mutant Models – CaMKII-NMDAR interactions in Learning and Memory: A Case Study in Ethical and Reproducible Modelling Approaches</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./publications.html" rel="next">
<link href="./40-model-validation.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-d4d76bf8491c20bad77d141916dc28e1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-0a0063f14e87237348a3db0020d9102a.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./40-results.html"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Analysis of CaMKII Dynamics in Wild-Type and Mutant Models</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">CaMKII-NMDAR interactions in Learning and Memory: A Case Study in Ethical and Reproducible Modelling Approaches</a> 
        <div class="sidebar-tools-main">
    <a href="./CaMKII-NMDAR-interactions-in-Learning-and-Memory--A-Case-Study-in-Ethical-and-Reproducible-Modelling-Approaches.pdf" title="Download PDF" class="quarto-navigation-tool px-1" aria-label="Download PDF"><i class="bi bi-file-pdf"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Declaration</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./10-intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introduction</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./20-intro-biology.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Biological Research Background</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./30-modelling_background.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Computational Modelling Background</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./20-repro_open.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Open, Reproducible and Ethics-focused PhD</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./20-data-hazards.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">A Case Study in Ethical Reflection: Data Hazards in this PhD</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./30-model.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">In Silico Model of CaMKII, NMDARs, and Associated Signalling Molecules</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./40-model-validation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Validating model behaviour</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./40-results.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Analysis of CaMKII Dynamics in Wild-Type and Mutant Models</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./publications.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">List of publications and presentations resulting from this PhD</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#wild-type-model" id="toc-wild-type-model" class="nav-link active" data-scroll-target="#wild-type-model"><span class="header-section-number">7.1</span> Wild-type model</a>
  <ul class="collapse">
  <li><a href="#calcium-release-and-initial-cam-saturation" id="toc-calcium-release-and-initial-cam-saturation" class="nav-link" data-scroll-target="#calcium-release-and-initial-cam-saturation"><span class="header-section-number">7.1.1</span> Calcium release and initial CaM saturation</a></li>
  <li><a href="#cam-stabilises-camkii-in-the-open-conformation-but-is-not-required-for-camkiinmdar-binding" id="toc-cam-stabilises-camkii-in-the-open-conformation-but-is-not-required-for-camkiinmdar-binding" class="nav-link" data-scroll-target="#cam-stabilises-camkii-in-the-open-conformation-but-is-not-required-for-camkiinmdar-binding"><span class="header-section-number">7.1.2</span> CaM stabilises CaMKII in the open conformation, but is not required for CaMKII/NMDAR binding</a></li>
  <li><a href="#camkii-subunits-remain-open-and-autophosphorylate-at-t286-upon-nmdar-binding-even-in-the-absence-of-cam" id="toc-camkii-subunits-remain-open-and-autophosphorylate-at-t286-upon-nmdar-binding-even-in-the-absence-of-cam" class="nav-link" data-scroll-target="#camkii-subunits-remain-open-and-autophosphorylate-at-t286-upon-nmdar-binding-even-in-the-absence-of-cam"><span class="header-section-number">7.1.3</span> CaMKII subunits remain open and autophosphorylate at T286 upon NMDAR binding, even in the absence of CaM</a></li>
  <li><a href="#camkii-phosphorylation-at-t286-is-not-more-likely-when-cam-is-present" id="toc-camkii-phosphorylation-at-t286-is-not-more-likely-when-cam-is-present" class="nav-link" data-scroll-target="#camkii-phosphorylation-at-t286-is-not-more-likely-when-cam-is-present"><span class="header-section-number">7.1.4</span> CaMKII phosphorylation at T286 is not more likely when CaM is present</a></li>
  <li><a href="#nmdar-binding-promotes-camkii-autophosphorylation-at-t286-and-t306-sites" id="toc-nmdar-binding-promotes-camkii-autophosphorylation-at-t286-and-t306-sites" class="nav-link" data-scroll-target="#nmdar-binding-promotes-camkii-autophosphorylation-at-t286-and-t306-sites"><span class="header-section-number">7.1.5</span> NMDAR binding promotes CaMKII autophosphorylation at T286 and T306 sites</a></li>
  </ul></li>
  <li><a href="#t286-and-camkiinmdar-binding-mt-models" id="toc-t286-and-camkiinmdar-binding-mt-models" class="nav-link" data-scroll-target="#t286-and-camkiinmdar-binding-mt-models"><span class="header-section-number">7.2</span> T286 and CaMKII/NMDAR binding MT models</a></li>
  <li><a href="#discussion-of-results" id="toc-discussion-of-results" class="nav-link" data-scroll-target="#discussion-of-results"><span class="header-section-number">7.3</span> Discussion of results</a></li>
  <li><a href="#limitations-and-future-steps" id="toc-limitations-and-future-steps" class="nav-link" data-scroll-target="#limitations-and-future-steps"><span class="header-section-number">7.4</span> Limitations and future steps</a></li>
  <li><a href="#section" id="toc-section" class="nav-link" data-scroll-target="#section"><span class="header-section-number">7.5</span> &lt;!–</a></li>
  <li><a href="#camkiinmdar-binding-mt-model" id="toc-camkiinmdar-binding-mt-model" class="nav-link" data-scroll-target="#camkiinmdar-binding-mt-model"><span class="header-section-number">7.6</span> CaMKII/NMDAR binding MT model</a></li>
  <li><a href="#statssss" id="toc-statssss" class="nav-link" data-scroll-target="#statssss"><span class="header-section-number">8</span> STATSSSS</a></li>
  <li><a href="#limitations" id="toc-limitations" class="nav-link" data-scroll-target="#limitations"><span class="header-section-number">9</span> Limitations</a></li>
  <li><a href="#discussion-of-results-1" id="toc-discussion-of-results-1" class="nav-link" data-scroll-target="#discussion-of-results-1"><span class="header-section-number">10</span> discussion of results</a></li>
  <li><a href="#future-directions-sec-future-directions" id="toc-future-directions-sec-future-directions" class="nav-link" data-scroll-target="#future-directions-sec-future-directions"><span class="header-section-number">11</span> Future directions [#sec-future-directions]</a>
  <ul class="collapse">
  <li><a href="#camkii-dodecamer" id="toc-camkii-dodecamer" class="nav-link" data-scroll-target="#camkii-dodecamer"><span class="header-section-number">11.1</span> CaMKII dodecamer</a></li>
  <li><a href="#in-camkii286p-mt" id="toc-in-camkii286p-mt" class="nav-link" data-scroll-target="#in-camkii286p-mt"><span class="header-section-number">11.2</span> in camkii286p MT</a>
  <ul class="collapse">
  <li><a href="#does-this-model-reproduce-previously-suggested-results-of-camkii-activation" id="toc-does-this-model-reproduce-previously-suggested-results-of-camkii-activation" class="nav-link" data-scroll-target="#does-this-model-reproduce-previously-suggested-results-of-camkii-activation"><span class="header-section-number">11.2.1</span> Does this model reproduce previously suggested results of CaMKII activation?</a></li>
  <li><a href="#camkii-interaction-with-nmdars" id="toc-camkii-interaction-with-nmdars" class="nav-link" data-scroll-target="#camkii-interaction-with-nmdars"><span class="header-section-number">11.2.2</span> CaMKII interaction with NMDARs</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#same-as-before-most-activity-of-camkii-is-from-autonomous-not-due-to-bound-to-cam" id="toc-same-as-before-most-activity-of-camkii-is-from-autonomous-not-due-to-bound-to-cam" class="nav-link" data-scroll-target="#same-as-before-most-activity-of-camkii-is-from-autonomous-not-due-to-bound-to-cam"><span class="header-section-number">12</span> same as before, most activity of camkii is from autonomous, not due to bound to cam:</a></li>
  <li><a href="#t305306-p" id="toc-t305306-p" class="nav-link" data-scroll-target="#t305306-p"><span class="header-section-number">13</span> t305/306 p</a>
  <ul class="collapse">
  <li><a href="#parameter-accuracy" id="toc-parameter-accuracy" class="nav-link" data-scroll-target="#parameter-accuracy"><span class="header-section-number">13.1</span> Parameter accuracy</a></li>
  </ul></li>
  <li><a href="#trying-to-understand-how-parameters-are-limiting-or-not" id="toc-trying-to-understand-how-parameters-are-limiting-or-not" class="nav-link" data-scroll-target="#trying-to-understand-how-parameters-are-limiting-or-not"><span class="header-section-number">14</span> Trying to understand how parameters are limiting or not</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span id="sec-results" class="quarto-section-identifier"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Analysis of CaMKII Dynamics in Wild-Type and Mutant Models</span></span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>Three principal model versions of the CaMKII/NMDAR model have been developed. The first is a wild-type (WT) model, as described in <a href="#sec-model-description" class="quarto-xref"><span class="quarto-unresolved-ref">sec-model-description</span></a>. In addition, two mutant (MT) models were constructed: one in which CaMKII is unable to phosphrylate at T286, and another in which CaMKII cannot bind to NMDARs. These mutations were implemented by setting the reaction rates corresponding to CaMKII autophosphorylation and CaMKII–NMDAR binding, respectively, to zero.</p>
<section id="wild-type-model" class="level2" data-number="7.1">
<h2 data-number="7.1" class="anchored" data-anchor-id="wild-type-model"><span class="header-section-number">7.1</span> Wild-type model</h2>
<p>This section presents a summary of the results obtained from the wild-type (WT) model. In the WT simulation, the final total number of CaMKII_open molecules reaches a mean of approximately 108 molecules at 100 seconds. The shaded area surrounding the mean trajectory in the graph represents the standard deviation (SD), indicating the degree of variability across different simulation runs.</p>
<p>Figure 7.2 provides an overview of the key molecular interactions and states observed in the wild-type (WT) CaMKII/NMDAR model during the simulation. Each subpanel (a–d) captures specific aspects of the molecular dynamics, focusing particularly on the interplay between CaMKII opening, T286 phosphorylation, CaM binding, and NMDAR binding.</p>
<p>Figure 7.2a shows the evolution of the open conformation of CaMKII over time. The brown line represents the total number of CaMKII_open subunits, which steadily increases as the simulation progresses. Two main factors contribute to the stabilization of CaMKII in the open state: (i) CaM binding (light blue line, camkii_cam_unbound_open) and (ii) NMDAR binding (dark blue line, nmdar_camkii_complex). Notably, NMDAR binding can maintain CaMKII in the open conformation even in the absence of CaM binding (Fig 7.2c). Figure 7.2b focuses on T286 phosphorylation dynamics of CaMKII. The orange line (camkii_t286p) shows the total number of CaMKII subunits phosphorylated at T286, while the pink line (camkii_t286p1_bound_nmdar) represents the subset of T286-phosphorylated subunits that are also bound to NMDARs. A large fraction of phosphorylated subunits are found associated with NMDARs, indicating that NMDAR binding promotes or sustains T286 phosphorylation. Importantly, this phosphorylation happens largely independently of CaM, as further illustrated in panels (c) and (d).</p>
<p>Figure 7.2c dissects the T286-phosphorylated CaMKII subunits that are bound to NMDARs into CaM-bound and CaM-unbound populations. Here, the pink line represents all T286-phosphorylated CaMKII subunits bound to NMDARs, while the orange line shows the subset that are CaM-free (camkii_cam_unbound_t286p1_bound_nmdar). It is clear that the majority of the T286-phosphorylated CaMKII subunits bound to NMDARs are unbound from CaM, emphasizing that CaMKII can maintain phosphorylation at T286 even after CaM dissociates. This further supports the idea that NMDAR binding alone is sufficient to stabilize the active, open, phosphorylated state in this model. Figure 7.2d provides a more detailed look at T286 phosphorylation in relation to CaM binding across the whole CaMKII population. The data shows that roughly equal numbers of CaMKII subunits are phosphorylated at T286 whether they are CaM-bound or CaM-unbound. This demonstrates that although CaM binding facilitates CaMKII opening and activation, T286 phosphorylation does not require continuous CaM binding once CaMKII is locked in its open, phosphorylated conformation.</p>
<p>To further dissect the molecular events underlying these dynamics, the following sections will explore a representative simulation run in detail, focusing on the specific behaviours of CaMKII subunits in relation to NMDAR and CaM binding, and T286 phosphorylation.</p>
<p>which figure is best? separated or all in one?</p>
<div id="fig-all-means-WT" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="Final value means of total molecule counts">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-all-means-WT-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="40-results-figures\WT\wt-redo.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1" title="Figure&nbsp;7.1: Final value means of total molecule counts (mean ± SD): camkii_open (108.00 ± 5.12); camkii_cam_ca4 (79.22 ± 3.85); nmdar_free (0.00 ± 0.00); nmdar_camkii_complex (30.00 ± 0.00); camkii_t286p (50.56 ± 5.96); camkii_t286p1_bound_nmdar (27.11 ± 1.37); camkii_cam_unbound_t286p1_bound_nmdar (24.67 ± 1.25); camkii_cam_ca4_t286p1 (24.33 ± 4.45); camkii_cam_unbound_open_t286p1 (26.22 ± 2.35)"><img src="40-results-figures\WT\wt-redo.png" class="img-fluid figure-img" data-fig-scap="Final value means of total molecule counts"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-all-means-WT-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7.1: Final value means of total molecule counts (mean ± SD): camkii_open (108.00 ± 5.12); camkii_cam_ca4 (79.22 ± 3.85); nmdar_free (0.00 ± 0.00); nmdar_camkii_complex (30.00 ± 0.00); camkii_t286p (50.56 ± 5.96); camkii_t286p1_bound_nmdar (27.11 ± 1.37); camkii_cam_unbound_t286p1_bound_nmdar (24.67 ± 1.25); camkii_cam_ca4_t286p1 (24.33 ± 4.45); camkii_cam_unbound_open_t286p1 (26.22 ± 2.35)
</figcaption>
</figure>
</div>
<div class="landscape">
<div id="fig-t286p-means" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="abcd">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-t286p-means-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="40-results-figures\WT\t286p-means.PNG" class="lightbox" data-gallery="quarto-lightbox-gallery-2" title="Figure&nbsp;7.2: Final value means of total molecule counts (mean ± SD): camkii_open (108.00 ± 5.12); camkii_cam_ca4 (79.22 ± 3.85); nmdar_free (0.00 ± 0.00); nmdar_camkii_complex (30.00 ± 0.00); camkii_t286p (50.56 ± 5.96); camkii_t286p1_bound_nmdar (27.11 ± 1.37); camkii_cam_unbound_t286p1_bound_nmdar (24.67 ± 1.25); camkii_cam_ca4_t286p1 (24.33 ± 4.45); camkii_cam_unbound_open_t286p1 (26.22 ± 2.35)"><img src="40-results-figures\WT\t286p-means.PNG" class="img-fluid figure-img" data-fig-scap="abcd"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-t286p-means-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7.2: Final value means of total molecule counts (mean ± SD): camkii_open (108.00 ± 5.12); camkii_cam_ca4 (79.22 ± 3.85); nmdar_free (0.00 ± 0.00); nmdar_camkii_complex (30.00 ± 0.00); camkii_t286p (50.56 ± 5.96); camkii_t286p1_bound_nmdar (27.11 ± 1.37); camkii_cam_unbound_t286p1_bound_nmdar (24.67 ± 1.25); camkii_cam_ca4_t286p1 (24.33 ± 4.45); camkii_cam_unbound_open_t286p1 (26.22 ± 2.35)
</figcaption>
</figure>
</div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="40-results-figures\observable_names.PNG" class="lightbox" data-gallery="quarto-lightbox-gallery-3" title="blablaneeds"><img src="40-results-figures\observable_names.PNG" class="img-fluid figure-img" alt="blablaneeds"></a></p>
<figcaption>blablaneeds</figcaption>
</figure>
</div>
<section id="calcium-release-and-initial-cam-saturation" class="level3" data-number="7.1.1">
<h3 data-number="7.1.1" class="anchored" data-anchor-id="calcium-release-and-initial-cam-saturation"><span class="header-section-number">7.1.1</span> Calcium release and initial CaM saturation</h3>
<p>In the WT model (described in detail in <a href="#sec-model-description" class="quarto-xref"><span class="quarto-unresolved-ref">sec-model-description</span></a>), a release of 1000 calcium ions occurs within a dendritic spine volume of 0.50588 <span class="math inline">\(\mu\text{m}^3\)</span>. This calcium subsequently binds to CaM, which can bind up to four calcium ions. Full saturation of CaM is required for subsequent binding to CaMKII, resulting in the formation of the CaMKII–CaM complex (CaMKII_CaM_Ca4).</p>
<p>This progression is illustrated in <a href="#fig-calcium-binding" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-calcium-binding</span></a>, where the gradual formation of partially and fully calcium-bound CaM species is shown. Although calcium itself is not displayed (due to its relatively high concentration distorting the plot scale), the stepwise increase in CaM species bound to one, two, three, and four calcium ions is clearly visible. The appearance of CaMKII bound to fully saturated CaM (camkii_cam_ca4) further confirms the expected dynamics. as per the set reaction rates and established literature, CaM fully bound to four calcium ions is immediately available for binding to CaMKII, whereas CaM bound to fewer calcium ions is less readily available for this interaction.</p>
<p>The simulation reveals a rapid decrease in the concentration of free CaM, which stabilises at approximately half of its initial value, suggesting that roughly half of the CaM molecules remain unbound to calcium. Upon further investigation, this suggests that calcium is acting as a saturating factor. Given that each CaM molecule can bind up to four calcium ions, the 290 CaM molecules would require at least 1200 calcium ions (290 × 4 <span class="math inline">\(\approx\)</span> 1200) for full saturation. However, the model uses 1000 calcium ions, which is slightly below this threshold. It is also worth noting that calcium ions are continuously binding to and dissociating from CaM as part of the ongoing reactions, so as some calcium may dissociate from one species, it might then associate to another, allowing for dynamic and emergent molecule behaviours. Even if the calcium concentration were increased (the difference between 1000 and 1200 calcium ions would be relatively small, from 3.282 μM to 3.938 μM, respectively), we would expect an increase in the number of CaM molecules bound to four calcium ions, but the change might not be substantial.</p>
<div id="fig-calcium-binding" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="CaM binding calcium" data-fig-pos="H">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-calcium-binding-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="40-results-figures\WT\calcium-binding.png" class="lightbox" data-gallery="quarto-lightbox-gallery-4" title="Figure&nbsp;7.3: CaM binding to 1,2,3 and 4 calcium ions (calcium not shown) and formation of CaMKII-CaM complex (CaMKII_CaM_Ca4). For reproducibility, this figure was obtained from plotting “run_2025-04-03_08-45-21_seed_2.”"><img src="40-results-figures\WT\calcium-binding.png" class="img-fluid figure-img" data-fig-pos="H" data-fig-scap="CaM binding calcium"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-calcium-binding-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7.3: CaM binding to 1,2,3 and 4 calcium ions (calcium not shown) and formation of CaMKII-CaM complex (CaMKII_CaM_Ca4). <em>For reproducibility, this figure was obtained from plotting “run_2025-04-03_08-45-21_seed_2.”</em>
</figcaption>
</figure>
</div>
</section>
<section id="cam-stabilises-camkii-in-the-open-conformation-but-is-not-required-for-camkiinmdar-binding" class="level3" data-number="7.1.2">
<h3 data-number="7.1.2" class="anchored" data-anchor-id="cam-stabilises-camkii-in-the-open-conformation-but-is-not-required-for-camkiinmdar-binding"><span class="header-section-number">7.1.2</span> CaM stabilises CaMKII in the open conformation, but is not required for CaMKII/NMDAR binding</h3>
<p>Next, we examine CaM–CaMKII binding (<a href="#fig-cam-camkii-binding" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-cam-camkii-binding</span></a>), focusing on how CaMKII’s open conformation of subunits stabilises over time. Out of the 720 CaMKII subunits released (not shown), a total of 104 molecules remain in the open conformation (camkii_open), at 100 seconds. Camkii_open represent the total population of CaMKII subunits that are structurally in their open state. To determine how many of these open subunits are simultaneously bound to CaM, we refer to the pink trace in <a href="#fig-cam-camkii-binding" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-cam-camkii-binding</span></a>, which indicates that 79 subunits are CaM-bound by the end of the simulation. This shows that approximately 75% of the open CaMKII subunits are being stabilised in this state through CaM binding up until the end of the simulation.</p>
<p>In contrast, the population of open CaMKII subunits that are not bound to CaM reaches equilibrium just before the 20-second mark, maintaining a steady count of just above 20 molecules until the end of the simulation. This accounts for the remaining 25% of the open CaMKII pool. These observations indicate that the majority of open CaMKII subunits are stabilised in their active conformation via CaM binding, while a smaller proportion is maintained in the open state by alternative mechanisms.</p>
<p>In the model at hand, CaMKII is defined as dynamically transitioning between open and closed conformations, and unless stabilised in the open state —either through CaM binding, T286 phosphorylation, or NMDAR association— it will revert to the closed state (see <a href="#fig-tbl-1" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-tbl-1</span></a>). So we can see in <a href="#fig-cam-camkii-binding" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-cam-camkii-binding</span></a> that the majority of CaMKII open is stabilised by CaM, however, there is another mechanism that is keeping CaMKII open. This other mechanism is in direct relation with NMDAR interaction as we will explore next.</p>
<p>By examining <a href="#fig-camkii-opening-nmdar" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-camkii-opening-nmdar</span></a>, we observe that the levels of CaMKII binding to NMDARs (dark blue line: nmdar_camkii_complex) closely follow the same pattern as CaM unbinding from CaMKII (medium blue: camkii_cam_unbound_open). In other words, as CaMKII associates with NMDARs, this interaction maintains CaMKII in an open state, without the need for CaM binding to CaMKII.</p>
<p>Interestingly, the proportion of CaMKII free from CaM that are still in their open conformation (medium blue: camkii_cam_unbound_open, <a href="#fig-camkii-opening-nmdar" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-camkii-opening-nmdar</span></a>) fluctuates up and down frequently, which comes from CaMKII opening and closing (as CaM is not bound yet).</p>
<p>Initially, the level of CaMKII free from CaM that are still in their open conformation (camkii_cam_unbound_open) fluctuates and exceeds the amount of CaMKII bound to NMDARs (nmdar_camkii_complex). This indicates that while a pool of open CaMKII is available, not all of it is bound to NMDARs at once. Over time, as CaMKII binding to NMDARs progresses and the system approaches a steady state, newly opened CaMKII subunits are quickly captured by available NMDAR binding sites. Consequently, the amount of free, open CaMKII stabilizes at a level consistently lower than the amount of NMDAR-bound CaMKII, indicating that binding to NMDARs efficiently sequesters and stabilizes open CaMKII in the later phase.</p>
<p>Moreover, in <a href="#fig-camkii-opening-nmdar" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-camkii-opening-nmdar</span></a>, we observe a three-step, ladder-like progression of CaMKII binding to NMDARs. Each step corresponds to an increase in the number of CaMKII subunits bound: first reaching approximately 12 subunits, then 24, and finally 30. This pattern suggests that entire CaMKII holoenzymes (each containing 12 subunits) sequentially bind to NMDARs.</p>
<p>Initially, one holoenzyme binds rapidly, with its subunits quickly associating with available NMDARs. Once 12 subunits are bound, a second holoenzyme begins binding. However, this process is slower, likely because after the first 12 subunits are bound, there are fewer free NMDARs left. Now, the second holoenzyme has fewer choices, and it may need to “search” or “diffuse” around more to find free receptors. Finally, after the second holoenzyme binds (up to 24 subunits), a third holoenzyme binds partially, occupying the remaining 6 NMDARs. By the time the third holoenzyme tries to bind, only 6 NMDARs are free — not enough for all 12 subunits. The progressive slowing reflects decreasing availability of free NMDARs for binding. The implications of these findings are discussed in the REFERENCE-SECTION; however, this particular mode of CaMKII binding to NMDARs may be an artifact of the model. Although not necessarily invalid, this aspect warrants further scrutiny and should be revisited in future research.</p>
<p>In summary, the majority (75%) of CaMKII subunits in the open conformation are stabilized by binding to CaM. However, while CaM helps stabilize CaMKII in the open state, the binding of CaMKII to NMDARs is not dependent on CaM. As CaMKII binds to NMDARs, it maintains its open conformation even without CaM binding, indicating that the interaction between CaMKII and NMDARs can stabilize the open state of CaMKII independently of CaM.</p>
<div id="fig-camkii-open" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-camkii-open-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-camkii-open" style="flex-basis: 100.0%;justify-content: flex-start;">
<div id="fig-cam-camkii-binding" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-cam-camkii-binding-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="40-results-figures\WT\cam-camkii-binding.png" class="lightbox" data-gallery="fig-camkii-open" title="Figure&nbsp;7.4&nbsp;(a): "><img src="40-results-figures\WT\cam-camkii-binding.png" id="fig-cam-camkii-binding" class="img-fluid figure-img" data-ref-parent="fig-camkii-open"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-cam-camkii-binding-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a)
</figcaption>
</figure>
</div>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-camkii-open" style="flex-basis: 100.0%;justify-content: flex-start;">
<div id="fig-camkii-opening-nmdar" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-camkii-opening-nmdar-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="40-results-figures\WT\camkii-open-nmdar.png" class="lightbox" data-gallery="fig-camkii-open" title="Figure&nbsp;7.4&nbsp;(b): "><img src="40-results-figures\WT\camkii-open-nmdar.png" id="fig-camkii-opening-nmdar" class="img-fluid figure-img" data-ref-parent="fig-camkii-open"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-camkii-opening-nmdar-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b)
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-camkii-open-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7.4: Stabilisation of CaMKII in the Open Conformation. <em>For reproducibility, this figure was obtained from plotting data from “run_2025-04-03_08-45-21_seed_2.”</em>
</figcaption>
</figure>
</div>
</section>
<section id="camkii-subunits-remain-open-and-autophosphorylate-at-t286-upon-nmdar-binding-even-in-the-absence-of-cam" class="level3" data-number="7.1.3">
<h3 data-number="7.1.3" class="anchored" data-anchor-id="camkii-subunits-remain-open-and-autophosphorylate-at-t286-upon-nmdar-binding-even-in-the-absence-of-cam"><span class="header-section-number">7.1.3</span> CaMKII subunits remain open and autophosphorylate at T286 upon NMDAR binding, even in the absence of CaM</h3>
<p>Next we look at total number of CaMKII subunits phosphorylated at T286. CaMKII_T286P reflects the full amount of T286-phosphorylated CaMKII, irrespective of any other concurrent molecular states. All CaMKII subunits categorised as camkii_t286p are necessarily in the open conformation. Therefore, camkii_t286p represents a subset of the camkii_open population <a href="#fig-subsets-of-open-t286p1_p0" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-subsets-of-open-t286p1_p0</span></a>. This is consistent with both the model’s coding and biological evidence, as phosphorylation at T286 locks CaMKII in an autonomously active, open conformation by preventing the regulatory segment from re-binding to the catalytic domain, which physically prevents closing of the subunit.</p>
<p>Out of the 104 CaMKII subunits that are open at the end of the simulation, 45 are found to be phosphorylated at T286 <a href="#fig-subsets-of-open-t286p1_p0" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-subsets-of-open-t286p1_p0</span></a>, and the rest (104 - 45 = 59) are therefore not phosphorylated at this site. This indicates that approximately half of the open CaMKII subunits are in a T286-phosphorylated state.</p>
<p>Of the 45 T286-phosphorylated CaMKII subunits, a subset of 25 are found to be bound to NMDARs (pink line: camkii_t286p1_bound_nmdar as shown in <a href="#fig-camkii-t286p-nmdar-b" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-camkii-t286p-nmdar-b</span></a>). The difference between these two molecules, corresponds to the number of CaMKII molecules that are T286-phosphorylated but not associated with NMDARs. This is, out of all the CaMKII subunits that are phosphorylated at T286, about half can be found bound to NMDARs, and the other half is not. Figure <a href="#fig-subsets_nmdar-t268p" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-subsets_nmdar-t268p</span></a> further demonstrates that the majority of CaMKII subunits bound to NMDARs and phosphorylated at T286 are unbound from CaM. This complements observations in <a href="#fig-camkii-t286p-nmdar-b" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-camkii-t286p-nmdar-b</span></a>, where 25 CaMKII subunits are both T286-phosphorylated and bound to NMDARs, of which 23 are CaM-free and only 2 remain CaM-bound by the end of the simulation.</p>
<p>Further, within the population of T286-phosphorylated CaMKII that is also bound to NMDARs (pink line: camkii_t286p1_bound_nmdar), only a very small number of subunits remain bound to CaM <a href="#fig-subsets_nmdar-t268p" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-subsets_nmdar-t268p</span></a>. Thus, the majority of CaMKII subunits that are bound in a NMDAR/CaMKII complex are both phosphorylated at T286 and free from CaM (which is consistent with literature, as further discussed in REFERENCE-SECTION). This is, once CaMKII is bound to the NMDARs, the holoenzyme subunits can remain open, and autophosphorylate even in the absence of CaM.</p>
<div id="fig-subsets-of-open-t286p1_p0" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="CaMKII open and T286 phosphorylated" data-fig-pos="H">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-subsets-of-open-t286p1_p0-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="40-results-figures\WT\subsets-of-open-t286p1_p0.png" class="lightbox" data-gallery="quarto-lightbox-gallery-7" title="Figure&nbsp;7.5: 1"><img src="40-results-figures\WT\subsets-of-open-t286p1_p0.png" class="img-fluid figure-img" data-fig-pos="H" data-fig-scap="CaMKII open and T286 phosphorylated"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-subsets-of-open-t286p1_p0-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7.5: 1
</figcaption>
</figure>
</div>
<div id="fig-camkii-t286p-nmdar-b" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="CaMKII T286 phosphorylated and bound to CaM" data-fig-pos="H">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-camkii-t286p-nmdar-b-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="40-results-figures\WT\camkii-t286p.png" class="lightbox" data-gallery="quarto-lightbox-gallery-8" title="Figure&nbsp;7.6: 2"><img src="40-results-figures\WT\camkii-t286p.png" class="img-fluid figure-img" data-fig-pos="H" data-fig-scap="CaMKII T286 phosphorylated and bound to CaM"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-camkii-t286p-nmdar-b-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7.6: 2
</figcaption>
</figure>
</div>
<div id="fig-subsets_nmdar-t268p" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="CaMKII T286 phosphorylated and bound to CaM" data-fig-pos="H">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-subsets_nmdar-t268p-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="40-results-figures\WT\subsets_nmdar-t268p.png" class="lightbox" data-gallery="quarto-lightbox-gallery-9" title="Figure&nbsp;7.7: alt text"><img src="40-results-figures\WT\subsets_nmdar-t268p.png" class="img-fluid figure-img" data-fig-pos="H" data-fig-scap="CaMKII T286 phosphorylated and bound to CaM"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-subsets_nmdar-t268p-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7.7: alt text
</figcaption>
</figure>
</div>
</section>
<section id="camkii-phosphorylation-at-t286-is-not-more-likely-when-cam-is-present" class="level3" data-number="7.1.4">
<h3 data-number="7.1.4" class="anchored" data-anchor-id="camkii-phosphorylation-at-t286-is-not-more-likely-when-cam-is-present"><span class="header-section-number">7.1.4</span> CaMKII phosphorylation at T286 is not more likely when CaM is present</h3>
<p>As explained in <a href="#sec-cam-binding-and-t286-p" class="quarto-xref"><span class="quarto-unresolved-ref">sec-cam-binding-and-t286-p</span></a>, we use a ‘conformational selection’ approach to modelling CaMKII activation, where CaMKII can exist in both open and closed states independently, and CaM selectively binds to and stabilizes the open conformation, without <em>inducing</em> a conformational change. In order to understand how CaM is interacting with CaMKII T286 phosphorylation, we look at <a href="#fig-subsets-of-camkii-t286p-cam-binding" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-subsets-of-camkii-t286p-cam-binding</span></a>. Here we can see that among all CaMKII subunits phosphorylated at T286, approximately half (23 molecules) are also bound to CaM, represented by the peach-coloured trace <em>“camkii_cam_ca4_t286p1”</em>, while the remaining half (22 molecules) are unbound from CaM (brown-coloured trace: <em>“camkii_cam_unbound_open_t286p1”</em>).</p>
<p>This indicates that, although CaM binding co-occurs with T286 phosphorylation, T286 is not necessarily more likely when subunits are bound to CaM. In fact, subunits that are both CaM-bound and T286-phosphorylated emerge more slowly during the simulation <a href="#fig-subsets-of-camkii-t286p-cam-binding" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-subsets-of-camkii-t286p-cam-binding</span></a>, suggesting that CaM binding is not the main event leading to the increase of initial CaMKII T286 phosphorylation, but as we will see next, NMDAR binding is (by keeping CaMKII subunits in their open conformation and allowing subsequent T286 autophosphorylation).</p>
<p>T286-phosphorylated CaMKII which is free from CaM (peach-coloured line in <a href="#fig-xsubsets_nmdar-t268p" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-xsubsets_nmdar-t268p</span></a>) corresponds in timing with the formation of the CaMKII/NMDAR complex <a href="#fig-subsets_nmdar-t268p" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-subsets_nmdar-t268p</span></a>. This suggests that, once CaMKII binds to NMDARs and its subunits are held in the open conformation, autophosphorylation at T286 is further facilitated in the CaMKII subunits that are part of the CaMKII/NMDAR complex.</p>
<p>Interestingly, a slight increase in the population of CaMKII free from CaM and T286 phosphorylayed is observed around 80 seconds. This rise suggests that subunits of holoenzymes locked into the CaMKII/NMDAR complex undergo further autophosphorylation of neighbouring subunits, resulting in a momentary increase in CaMKII subunits that are free from CaM and phosphorylated at T286.</p>
<div id="fig-subsets-of-camkii-t286p-cam-binding" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-subsets-of-camkii-t286p-cam-binding-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="40-results-figures/WT/subsets-of-camkii-t286p-cam-binding.png" class="lightbox" data-gallery="quarto-lightbox-gallery-10" title="Figure&nbsp;7.8: note for figure: its a redundancy that camkii_cam_unbound_optent28p1 has open in it because all camkii that is t286p1 is open anyway."><img src="40-results-figures/WT/subsets-of-camkii-t286p-cam-binding.png" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-subsets-of-camkii-t286p-cam-binding-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7.8: note for figure: its a redundancy that camkii_cam_unbound_optent28p1 has open in it because all camkii that is t286p1 is open anyway.
</figcaption>
</figure>
</div>
<div id="fig-xsubsets_nmdar-t268p" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-xsubsets_nmdar-t268p-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="40-results-figures\WT\xsubsets_nmdar-t268p.png" class="lightbox" data-gallery="quarto-lightbox-gallery-11" title="Figure&nbsp;7.9: 1"><img src="40-results-figures\WT\xsubsets_nmdar-t268p.png" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-xsubsets_nmdar-t268p-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7.9: 1
</figcaption>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="40-results-figures\WT\subset-of-nmdar-p-not-pt286.png" class="lightbox" data-gallery="quarto-lightbox-gallery-12" title="1.1"><img src="40-results-figures\WT\subset-of-nmdar-p-not-pt286.png" class="img-fluid figure-img" alt="1.1"></a></p>
<figcaption>1.1</figcaption>
</figure>
</div>
<div id="fig-xsubsets_nmdar-t268p0" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-xsubsets_nmdar-t268p0-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="40-results-figures\WT\camkii-nmdar-unphospho.png" class="lightbox" data-gallery="quarto-lightbox-gallery-13" title="Figure&nbsp;7.10: 2.2"><img src="40-results-figures\WT\camkii-nmdar-unphospho.png" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-xsubsets_nmdar-t268p0-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7.10: 2.2
</figcaption>
</figure>
</div>
</section>
<section id="nmdar-binding-promotes-camkii-autophosphorylation-at-t286-and-t306-sites" class="level3" data-number="7.1.5">
<h3 data-number="7.1.5" class="anchored" data-anchor-id="nmdar-binding-promotes-camkii-autophosphorylation-at-t286-and-t306-sites"><span class="header-section-number">7.1.5</span> NMDAR binding promotes CaMKII autophosphorylation at T286 and T306 sites</h3>
<p>Phosphorylation at the T306 site is known to occur at a slower rate than phosphorylation at T286 <span class="citation" data-cites="otmakhov2015Fast chang2017CaMKII">(<a href="#ref-otmakhov2015Fast" role="doc-biblioref">Otmakhov, Regmi, and Lisman 2015</a>; <a href="#ref-chang2017CaMKII" role="doc-biblioref">Chang et al. 2017</a>)</span>, and this is reflected in the model outputs. A total of 25 CaMKII subunits become phosphorylated at T306 over the course of the simulation (marroon trace: camkii_t306p in <a href="#fig-t306p" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-t306p</span></a>), of which 21 are bound to the NMDAR (golden trace:camkii_cam_unbound_t306p1_bound_nmdar in <a href="#fig-t306p" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-t306p</span></a>) —indicating that approximately 84% of T306-phosphorylated subunits are part of NMDAR complexes.</p>
<p>Moreover, the accumulation of doubly phosphorylated subunits (T286P and T306P; here referred to as CaMKII_PP, hot pink trace in <a href="#fig-t306p" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-t306p</span></a>) appears to track closely with the rate of T306 phosphorylation. This indicates that T306 phosphorylation typically follows T286 phosphorylation, particularly during the earlier stages of the simulation. This behaviour is consistent with both the modelled mechanisms and findings reported in the literature [REFERENCE].</p>
<div id="fig-t306p" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-pos="H">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-t306p-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="40-results-figures\WT\t306p.png" class="lightbox" data-gallery="quarto-lightbox-gallery-14" title="Figure&nbsp;7.11: alt text"><img src="40-results-figures\WT\t306p.png" class="img-fluid figure-img" data-fig-pos="H"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-t306p-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7.11: alt text
</figcaption>
</figure>
</div>
<p>Interestingly, after around 60 seconds, subunits that are solely phosphorylayed at T306 continue to increase, whilst CaMKII_PP does not (<a href="#fig-t306p" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-t306p</span></a>). This suggests that, at later stages in the simulation, T306 phosphorylation can occur and persist independently of T286 phosphorylation. Extending the simulation time may offer further insights into this dynamic, as it could support the biological hypothesis that initial T286 phosphorylation enables subsequent T306 phosphorylation, but that T306P can then be maintained even in the absence of T286P [REFERENCE].</p>
<p>Furthermore, T306 phosphorylation follows NMDAR binding dynamics closely. As shown in <a href="#fig-t306p-nmdar-binding" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-t306p-nmdar-binding</span></a>, CaMKII subunits initially bind to NMDARs in an unphosphorylated T306 state (slate-green: camkii_t306p0_bound_nmdar in <a href="#fig-t306p-nmdar-binding" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-t306p-nmdar-binding</span></a>). Once the CaMKII–NMDAR complex reaches equilibrium, levels of T306 phosphorylation increase correspondingly, as unphosphorylated T306 decreases.</p>
<div id="fig-t306p-nmdar-binding" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-pos="H">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-t306p-nmdar-binding-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="40-results-figures\WT\t306p-nmdar-binding.png" class="lightbox" data-gallery="quarto-lightbox-gallery-15" title="Figure&nbsp;7.12: alt text"><img src="40-results-figures\WT\t306p-nmdar-binding.png" class="img-fluid figure-img" data-fig-pos="H"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-t306p-nmdar-binding-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7.12: alt text
</figcaption>
</figure>
</div>
<p>In the final phase of CaMKII binding to NMDARs, after approximately 40 seconds —when all NMDARs are saturated, T306 phosphorylation continues to rise. This is likely supported by the concurrent increase in T286 phosphorylation, and by a decrease in CaM binding to CaMKII subunits already bound to NMDARs. That is, by 40s, a greater proportion of CaMKII subunits are free from CaM (as seen in <a href="#fig-subsets-of-camkii-t286p-cam-binding" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-subsets-of-camkii-t286p-cam-binding</span></a>). This aligns with expectations, as CaM binding and T306 phosphorylation are mutually exclusive processes —CaM must be unbound for T306 phosphorylation to occur. Therefore, as the model shows a decrease in CaM binding to CaMKII subunits that are already part of the CaMKII–NMDAR complex, the opportunity for T306 phosphorylation increases. In other words, the less CaM is bound to CaMKII subunits, the more likely it is that T306 can become phosphorylated, and the less likely CaM is to bind.</p>
<p><a href="#fig-t306p-nmdar-binding" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-t306p-nmdar-binding</span></a> shows three distinct peaks, each representing CaMKII subunits that are in the “p00” state — meaning they are not phosphorylated at either T286 or T306. This pattern highlights three important insights. First, CaMKII binds to NMDARs before any phosphorylation has occurred. Second, although phosphorylation is not required for binding, once CaMKII is incorporated into an NMDAR complex, autophosphorylation becomes more likely. Third, phosphorylation at T306 follows phosphorylation at T286, following a sequential phosphorylation mechanism as expected.</p>
<p>These findings show that the formation of CaMKII-NMDAR complexes does not depend on prior phosphorylation at T286. However, once T286 phosphorylation occurs, it can initiate a cascade of further autophosphorylation events, including at T306, within the same NMDAR/CaMKII complex. Previous studies have shown that CaMKII does not need to be phosphorylated at T286 in order to bind to NMDARs or to perform its catalytic functions. That said, phosphorylation at T286 can modulate or enhance both the binding affinity and the enzymatic activity of CaMKII, making it an important regulatory step rather than an absolute requirement.</p>
</section>
</section>
<section id="t286-and-camkiinmdar-binding-mt-models" class="level2" data-number="7.2">
<h2 data-number="7.2" class="anchored" data-anchor-id="t286-and-camkiinmdar-binding-mt-models"><span class="header-section-number">7.2</span> T286 and CaMKII/NMDAR binding MT models</h2>
<p>The T286 MT model is simialr biologically to the CaMKII T286A MT, in which the critical autophosphorylation site threonine 286 is substituted with alanine, preventing phosphorylation at this site. This mutation has been extensively characterised in both <em>in vitro</em> biochemical assays and <em>in vivo</em> using mice that were genetically modified. T286A mutation has been shown to impair LTP and spatial learning in rodents. he computational implementation allows investigation into how the loss of T286 phosphorylation alters CaMKII dynamics and downstream NMDAR interactions under different stimulation conditions [REFERENCES]. In the T286 MT model, this mutation is represented by setting reaction rule #8 to zero (<a href="#fig-tbl-1" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-tbl-1</span></a>), so phosphorylation can not happen.</p>
<p>The NMDAR/CaMKII binding MT model simulates a mutation that disrupts the interaction between CaMKII and the GluN2B subunit of NMDARs. This disruption can be achieved through mutations in either CaMKII or GluN2B, preventing their binding without necessarily altering other functions such as kinase activity [REFERENCES]. A well-characterized example is the CaMKII I205K mutation, where isoleucine is replaced by lysine. This substitution significantly impairs binding to GluN2B while leaving kinase activity intact [REFERENCE]. Experimental evidence shows that this mutation also reduces CaMKII accumulation at synapses following LTP induction [REFERENCE]. CHECK - this would be good to see in vizz output and future models. In the CaMKII/NMDAR MT model, this binding disruption is represented by setting reaction rule #12 to zero (<a href="#fig-tbl-1" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-tbl-1</span></a>), so NMDAR and CaMKII binding can not proceed.</p>
<p>The following analysis compares wild-type (WT) simulations with those of the two MT models. The total number of simulation runs for each group is as follows: WT (N = 9), T286 MT (N = 9), and CaMKII/NMDAR MT (N = 4). t-test and one-way anovas were performed with prior testing for normality of data sets using whitely test.</p>
<p>Simulations were conducted under identical parameters across all conditions (see <a href="#fig-molecule-concentrations" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-molecule-concentrations</span></a> for initial molecule release and <a href="#fig-tbl-1" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-tbl-1</span></a> for reaction rates), except for two CaMKII/NMDAR MT runs, as I explain below, which were part of preliminary tests involving a different CaMKII release state. Aside from this, the only intended differences between groups were the modifications to the relevant reaction rules described above.</p>
<p>The reduced number of simulations for the CaMKII/NMDAR MT model resulted from a human error during file execution, which will be explored in greater detail in the discussion section below. Due to this error, only four usable simulation runs were available for this model. It is important to note that, while all other simulation groups released CaMKII in its open state, the CaMKII/NMDAR MT group includes two runs with CaMKII released in the closed state and two with CaMKII released in the open state. These files, originally generated during preliminary testing (data not shown), were the only available usable runs for this condition due to the aforementioned error. As no substantial differences in simulation outcomes were observed between the two release states, all four runs were retained in the analysis. This detail is provided here for transparency.</p>
<p>Having established the simulation conditions for each model, we now present quantitative comparisons between wild-type and MT simulations across key molecular outputs. A primary focus of this analysis is the extent of CaMKII opening, as this conformational state is influenced by both T286 phosphorylation and binding to NMDARs. Given that both of these interactions promote and stabilise the open conformation of CaMKII, it was expected that disrupting either mechanism, through mutation at T286 or impairment of NMDAR binding, would lead to a reduction in CaMKII opening.</p>
<p><a href="#fig-camkii_open_anova" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-camkii_open_anova</span></a> displays the results of an ANOVA, with an F-statistic of 36.327, indicating a substantial difference between the group means relative to the variation within each group. Further, the associated p-value of <span class="math inline">\(3.219 \times 10^{-7}\)</span> indicates a very low probability that the observed differences between the groups are due to random chance.</p>
<p>Both mutations significantly decreased CaMKII opening relative to the wild-type (WT) control. The reduction observed in the T286_MT group was modest but statistically significant (p &lt; 0.05), supporting the role of T286 phosphorylation in stabilising the open conformation of CaMKII. These results were expected, as abolishing T286 phosphorylation means decreasing the chances of CaMKII subunits being kept in their open conformation via this mechanism.</p>
<p>The NMDAR_CaMKII_MT group exhibited a much more pronounced and highly significant reduction in CaMKII opening compared to the wild-type (WT) group (p &lt; 0.0001). As expected, reduced binding to NMDARs corresponds to a decreased likelihood of CaMKII remaining in its open conformation. Interestingly, CaMKII opening in the NMDAR_CaMKII_MT group was not only significantly lower than in the WT group, but also lower than in the T286_MT group (p &lt; 0.0001). These findings reinforce the idea that NMDAR binding plays a central role in maintaining CaMKII in its open state in our simulations. While T286 phosphorylation is also important, it is the disruption of CaMKII/NMDAR binding that most strongly impacts CaMKII opening levels. &lt;!–&gt; <img src="40-results-figures\statistics\camkii_open_anova.png" id="fig-camkii_open_anova" class="img-fluid" data-fig-scap="Both the T286 MT and the NMDAR/CaMKII binding MT exhibit significantly lower levels of CaMKII opening compared to WT" alt="Both the T286 MT (T286_MT) and the NMDAR/CaMKII binding MT (NMDAR_CaMKII_MT) exhibit significantly lower levels of CaMKII opening compared to WT, with the largest reduction observed in the NMDAR_CaMKII_MT condition. Final molecule counts for camkii_open are shown for each simulation run (dots), with group-level distributions represented as boxplots. Solid lines represent medians, and dashed lines represent means. Statistical comparisons were performed using one-way ANOVA followed by Tukey’s HSD post hoc test. Asterisks indicate levels of statistical significance from Tukey’s HSD post hoc test: p < 0.05 (*), p < 0.001 (), and p < 0.0001 (**). ANOVA F-statistic between groups = 36.327, and p = 3.219 \times 10^{-7}."></p>
<p>Next, I aimed to examine whether CaM binding to CaMKII was affected across the three models. Given the expected reduction in CaMKII opening, I expected that this decrease might also result in reduced CaM binding in both MT models. Specifically, I wanted to explore how CaM binding is related to both T286 phosphorylation and NMDAR/CaMKII binding.</p>
<p>As shown in <a href="#fig-CaMKII_CaM_Ca4_anova" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-CaMKII_CaM_Ca4_anova</span></a>, total CaMKII bound to CaM (camkii_cam_ca4) was significantly reduced only in the T286 MT. The F-value of 4.668 indicates that the variance <em>between</em> groups (i.e., how much the CaM binding values differ between the WT, T286_MT, and NMDAR/CaMKII_MT) is considerably larger than the variance <em>within</em> groups (i.e., how much individual data points vary within each group). In other words, the groups are not just randomly different; the differences between the groups are meaningful and consistent. The significant reduction in CaM binding in the T286 MT (p &lt; 0.05), and the lack of difference in the NMDAR/CaMKII MT compared to WT, suggests that T286 phosphorylation influences CaM binding to CaMKII by keeping it in the open state. On the other hand, NMDAR binding does not appear to have a similar effect.</p>
<p>Although these findings were initially unexpected, as I had anticipated that reduced CaMKII opening in both MTs might affect CaM binding in both models, the results are actually consistent with the underlying mechanisms that we had previously obeserved. In the WT model, CaM binding and NMDAR binding to CaMKII subunits appeared to be independent processes. Notably, the majority of CaMKII bound to NMDARs did not interact with CaM. This independence between NMDAR binding and CaM binding to CaMKII helps explain why the NMDAR/CaMKII MT, which disrupts NMDAR binding, does not significantly affect CaM binding. Since NMDAR binding and CaM binding to CaMKII are independent events, disrupting NMDAR binding does not impact the ability of CaMKII to bind to or release CaM. In contrast, the T286 mutation does lead to a reduction in CaM binding, highlighting the specific role of T286 phosphorylation in regulating CaM binding.</p>
<div id="fig-CaMKII_CaM_Ca4_anova" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="CaM binding to CaMKII is significantly reduced only with the loss of T286 phosphorylation">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-CaMKII_CaM_Ca4_anova-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="40-results-figures\statistics\CaMKII_CaM_Ca4_anova.png" class="lightbox" data-gallery="quarto-lightbox-gallery-16" title="Figure&nbsp;7.13: CaM binding to CaMKII is significantly reduced only with the loss of T286 phosphorylation. Final molecule counts for camkii_cam_ca4 are shown for each simulation run (dots), with group-level distributions represented as boxplots. Solid lines represent medians, and dashed lines represent means. The T286 MT (T286_MT) shows a significant reduction in CaM-bound CaMKII compared to wild-type (WT) (p < 0.05), whereas no significant difference (‘ns’) is observed between WT and the NMDAR/CaMKII MT (NMDAR_CaMKII_MT), or between the two MT groups. Statistical analysis was performed using one-way ANOVA (F = 4.668, p = 0.022), followed by Tukey’s HSD post hoc test."><img src="40-results-figures\statistics\CaMKII_CaM_Ca4_anova.png" class="img-fluid figure-img" data-fig-scap="CaM binding to CaMKII is significantly reduced only with the loss of T286 phosphorylation"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-CaMKII_CaM_Ca4_anova-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7.13: CaM binding to CaMKII is significantly reduced only with the loss of T286 phosphorylation. Final molecule counts for camkii_cam_ca4 are shown for each simulation run (dots), with group-level distributions represented as boxplots. Solid lines represent medians, and dashed lines represent means. The T286 MT (T286_MT) shows a significant reduction in CaM-bound CaMKII compared to wild-type (WT) (p &lt; 0.05), whereas no significant difference (‘ns’) is observed between WT and the NMDAR/CaMKII MT (NMDAR_CaMKII_MT), or between the two MT groups. Statistical analysis was performed using one-way ANOVA (F = 4.668, p = 0.022), followed by Tukey’s HSD post hoc test.
</figcaption>
</figure>
</div>
<p>To further explore how NMDARs and T286 phosphorylation affect CaMKII opening, I looked at the amount of CaMKII subunits that remain free from CaM and are kept in their open state (denoted as CaMKII–CaM_unbound_open, <a href="#fig-camkii_cam_unbound_open" class="quarto-xref">Figure&nbsp;<span class="quarto-unresolved-ref">fig-camkii_cam_unbound_open</span></a>). I observed a significant decrease in this molecule count in the NMDAR/CaMKII binding MT. As discussed previously, in the WT model, NMDARs play an active role in stabilizing CaMKII in its open state, even in the absence of CaM. This means that, in this model, NMDARs are crucial for maintaining CaMKII in its open form, free from calmodulin. When this stabilizing mechanism is disrupted in the NMDAR/CaMKII mutant, less CaMKII remains in the open state without CaM, because the NMDAR binding, which normally helps keep the CaMKII subunits open and CaM-free, is no longer functioning. As a result, there is a reduction in the amount of CaMKII remaining in its open state. Conversely, in the T286 mutant model, the ability of NMDARs to keep CaMKII in its open, CaM-free state is still intact, which explains why this aspect remains unaffected in this model.</p>
<div id="fig-camkii_cam_unbound_open" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="NMDAR binding is crucial for maintaining CaMKII in its open, CaM-free state">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-camkii_cam_unbound_open-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="40-results-figures\statistics\camkii_cam_unbound_open.png" class="lightbox" data-gallery="quarto-lightbox-gallery-17" title="Figure&nbsp;7.14: NMDAR binding is crucial for maintaining CaMKII in its open, CaM-free state (camkii_cam_unbound_open). Boxplot shows final molecule counts for camkii_cam_unbound_open across wild-type (WT), T286 phosphorylation mutant (T286_MT), and NMDAR/CaMKII binding mutant (NMDAR_CaMKII_MT) models. Individual simulation runs are represented as dots. Boxes represent the interquartile range, solid lines show medians, and dashed lines show group means. A one-way ANOVA revealed a significant group effect (F = 236.180, p = 3.80 \times 10^{-14}), with post hoc Tukey’s HSD tests indicating significantly fewer CaMKII molecules in this state in the NMDAR/CaMKII mutant compared to both WT and T286_MT (**p < 0.001). No significant difference was observed between WT and T286_MT."><img src="40-results-figures\statistics\camkii_cam_unbound_open.png" class="img-fluid figure-img" data-fig-scap="NMDAR binding is crucial for maintaining CaMKII in its open, CaM-free state"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-camkii_cam_unbound_open-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7.14: NMDAR binding is crucial for maintaining CaMKII in its open, CaM-free state (camkii_cam_unbound_open). Boxplot shows final molecule counts for camkii_cam_unbound_open across wild-type (WT), T286 phosphorylation mutant (T286_MT), and NMDAR/CaMKII binding mutant (NMDAR_CaMKII_MT) models. Individual simulation runs are represented as dots. Boxes represent the interquartile range, solid lines show medians, and dashed lines show group means. A one-way ANOVA revealed a significant group effect (F = 236.180, <span class="math inline">\(p = 3.80 \times 10^{-14}\)</span>), with post hoc Tukey’s HSD tests indicating significantly fewer CaMKII molecules in this state in the NMDAR/CaMKII mutant compared to both WT and T286_MT (**p &lt; 0.001). No significant difference was observed between WT and T286_MT.
</figcaption>
</figure>
</div>
<p>Since NMDARs play an essential role in keeping CaMKII subunits open and free from CaM (as shown by the CaMKII_cam_unbound_open data), this opens up the question of whether T286 phosphorylation, which is a key event for CaMKII activation, is also influenced by NMDAR signalling.</p>
<p>suggesting that CaM binding is not the main event leading to the increase of initial CaMKII T286 phosphorylation, but as we will see next, NMDAR binding is (by keeping CaMKII subunits in their open conformation and allowing subsequent T286 autophosphorylation).</p>
<!-- 
![camkii_cam_ca4_variance](40-results-figures\statistics\camkii_cam_ca4_variance.png)

![CaMKII_CaM_Ca4_ttest](40-results-figures\statistics\CaMKII_CaM_Ca4_ttest.png)

![camkii_open_variance](40-results-figures\statistics\camkii_open_variance.png)

![camkii_open_ttest](40-results-figures\statistics\camkii_open_ttest.png)

![what does this mean](40-results-figures\statistics\cam_unbound_open.png)

![what does this mean](40-results-figures\statistics\cam_unbound_open_ttest.png)

![nmdar_MTvsWT. no t-test because final mean and variance are same between both sets. what else could I do here?](40-results-figures\statistics\nmdar_MTvsWT.png)
-->
</section>
<section id="discussion-of-results" class="level2" data-number="7.3">
<h2 data-number="7.3" class="anchored" data-anchor-id="discussion-of-results"><span class="header-section-number">7.3</span> Discussion of results</h2>
</section>
<section id="limitations-and-future-steps" class="level2" data-number="7.4">
<h2 data-number="7.4" class="anchored" data-anchor-id="limitations-and-future-steps"><span class="header-section-number">7.4</span> Limitations and future steps</h2>
</section>
<section id="section" class="level2" data-number="7.5">
<h2 data-number="7.5" class="anchored" data-anchor-id="section"><span class="header-section-number">7.5</span> &lt;!–</h2>
<p>The NMDAR/CaMKII binding MT model simulates a mutation that disrupts the interaction between CaMKII and the GluN2B subunit of NMDARs. This disruption can be achieved through mutations in either CaMKII or GluN2B, preventing their binding without necessarily altering other functions such as kinase activity [REFERENCES]. A well-characterized example is the CaMKII I205K mutation, where isoleucine is replaced by lysine. This substitution significantly impairs binding to GluN2B while leaving kinase activity intact [REFERENCE]. Experimental evidence shows that this mutation also reduces CaMKII accumulation at synapses following LTP induction [REFERENCE].</p>
<p>okayyyy, so becayse t286 phosphorylation increases the chances of camkii subunits remaining in their open conformation, abolishing t286 phosphorylation we would exxpect a decrease in this opening. so we would expect iverall decrease of total camkii_open - but contrary to these expactions, we don’t see that!</p>
</section>
<section id="camkiinmdar-binding-mt-model" class="level2" data-number="7.6">
<h2 data-number="7.6" class="anchored" data-anchor-id="camkiinmdar-binding-mt-model"><span class="header-section-number">7.6</span> CaMKII/NMDAR binding MT model</h2>
<p>This model represents a mutation that prevents CaMKII from binding to the GluN2B subunit of NMDARs, a key interaction implicated in synaptic plasticity, as discussed in SECTION-REFERENCE. When it comes to <em>in vivo</em> and <em>in vitro</em> MTs of CaMKII and NMDAR binding, the mutation can tackle either molecule to prevent their binding. For example, a CaMKII I205K mutation in which isoleucine is replaced with a lysine impairs binding to GluN2B without affecting kinase activity [REFERENCE], and is demonstrated to reduce CaMKII accumulation at synapses after LTP induction [REFERENCE].</p>
<p>example of nmdar MT https://link.springer.com/article/10.1186/1756-6606-6-10 impaired by the mutations L1298A and R1300Q [22]. https://www.embopress.org/doi/full/10.1038/emboj.2011.482</p>
<p>Why L1298 and R1300? These residues are located in a short, conserved sequence motif within the cytoplasmic tail of GluN2B (often referred to as the CaMKII-binding site), which spans approximately amino acids 1290–1309. Structural, biochemical, and mutational studies have pinpointed L1298 and R1300 as essential contact points for high-affinity binding to the CaMKII kinase domain. Leucine 1298 (L1298) A hydrophobic residue, crucial for fitting into a hydrophobic pocket on CaMKII.</p>
<p>Mutating it to alanine removes hydrophobic bulk, destabilising this core interaction.</p>
<p>This significantly reduces binding affinity without disrupting the structure of GluN2B broadly.</p>
<ol start="2" type="1">
<li>Arginine 1300 (R1300) A positively charged residue, likely involved in electrostatic interactions or hydrogen bonding with CaMKII residues.</li>
</ol>
<p>Substituting arginine with glutamine (neutral, polar) removes the charge and disrupts those interactions. In biological systems, this has been examined using GluN2B MTs that disrupt the CaMKII-binding motif, or CaMKII MTs that selectively impair binding without affecting kinase activity. Both in vitro studies and in vivo models have demonstrated that disrupting this interaction impairs LTP maintenance and learning-related plasticity. The computational model isolates this mechanism by setting CaMKII–NMDAR binding reactions to zero, allowing for a focused analysis of how this interaction contributes to model behaviour.</p>
</section>
<section id="statssss" class="level1" data-number="8">
<h1 data-number="8"><span class="header-section-number">8</span> STATSSSS</h1>
<p>NORMALITY TEST RESULTS:</p>
<p>camkii_open</p>
<p>WT: 9 values |mutant-t286: 9 values</p>
<p>📊 Shapiro–Wilk test for WT W = 0.9399, p = 0.5811 ✅ Data appears normally distributed.</p>
<p>📊 Shapiro–Wilk test formutant W = 0.8739, p = 0.1353 ✅ Data appears normally distributed.</p>
<p>MT -NMDAR :10</p>
<p>📊 Shapiro–Wilk test formutant W = 0.9458, p = 0.6189 ✅ Data appears normally distributed.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="40-results-figures\statistics\camkii_open2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-18" title="camkii_open"><img src="40-results-figures\statistics\camkii_open2.png" class="img-fluid figure-img" alt="camkii_open"></a></p>
<figcaption>camkii_open</figcaption>
</figure>
</div>
<p>ANOVA F-statistic: 8.498, p-value: 1.528e-03</p>
<p>Tukey’s HSD Post Hoc Test: Multiple Comparison of Means - Tukey HSD, FWER=0.05<br>
=============================================================== group1 group2 meandiff p-adj lower upper reject ————————————————————— NMDAR_CaMKII_MT T286_MT -8.5667 0.0023 -14.1775 -2.9558 True NMDAR_CaMKII_MT WT -0.9 0.9161 -6.5109 4.7109 False T286_MT WT 7.6667 0.0076 1.91 13.4233 True</p>
<p>no stat difference for t306p</p>
<p>ANOVA F-statistic: 1.493, p-value: 2.441e-01</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="40-results-figures\statistics\t396p.png" class="lightbox" data-gallery="quarto-lightbox-gallery-19" title="camkii_t306p"><img src="40-results-figures\statistics\t396p.png" class="img-fluid figure-img" alt="camkii_t306p"></a></p>
<figcaption>camkii_t306p</figcaption>
</figure>
</div>
<p>https://blog.4dcu.be/programming/2021/12/30/Posthoc-Statannotations.html</p>
<p>https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.tukey_hsd.html</p>
<p>for camkii_cam_ca4</p>
<p>Tukey’s HSD Post Hoc Test: Multiple Comparison of Means - Tukey HSD, FWER=0.05<br>
=============================================================== group1 group2 meandiff p-adj lower upper reject ————————————————————— NMDAR_CaMKII_MT T286_MT -6.7111 0.0042 -11.4067 -2.0155 True NMDAR_CaMKII_MT WT 0.6222 0.9419 -4.0734 5.3178 False T286_MT WT 7.3333 0.0024 2.5158 12.1509 True ————————————————————— (mcell4-env)</p>
</section>
<section id="limitations" class="level1" data-number="9">
<h1 data-number="9"><span class="header-section-number">9</span> Limitations</h1>
</section>
<section id="discussion-of-results-1" class="level1" data-number="10">
<h1 data-number="10"><span class="header-section-number">10</span> discussion of results</h1>
<p>future step: having fully bound CaM can be useful to test whether the system can even reach peak activity, and CaMKII can be phosphorylated at levels shown in literature.</p>
<p>. It is also important to understand whether the proposed constitutive enzymatic activity of such subunits leading to T286 phosphorylation is critical for maintaining the CaMKII-NMDAR complex. What is clear is that given the ∼40-fold higher concentration of CaMKII subunits than GluN2B subunits (see sects. 9.2.2 and 13.2), https://journals.physiology.org/doi/full/10.1152/physrev.00034.2022</p>
</section>
<section id="future-directions-sec-future-directions" class="level1" data-number="11">
<h1 data-number="11"><span class="header-section-number">11</span> Future directions [#sec-future-directions]</h1>
<p>to see if more t306 p leads to indirect higher camkii closing, run for longer: Notably, the sustained presence of T306P may contribute to the closure of CaMKII subunits, thereby reducing their catalytic activity. This aligns with hypotheses that link T306 phosphorylation to the functional inhibition of CaMKII. The observed co-occurrence of NMDAR binding and T306 phosphorylation in the majority of subunits further supports a mechanism by which these interactions contribute to reduced CaMKII activity. A competing hypothesis, discussed in the Reference Section, posits that NMDAR binding suppresses T306 phosphorylation; however, our model does not support this outcome under the simulated conditions.</p>
<p>Further developments of this model to account for a more complex NMDAR molecule are possible, as I propose in [DISCUSSION SECTION].</p>
<p>The combined phosphorylation of Thr286, Thr305, and Thr306 is considered vital to the mechanism of subunit exchange because the regulatory domain needs to be accessible to interact with negatively charged residues located on the hub at the vertical interface between subunits, leading to the release of dimers. https://www.proquest.com/docview/3165639837?pq-origsite=gscholar&amp;fromopenview=true&amp;sourcetype=Dissertations%20&amp;%20Theses</p>
<p>decrease volume increase calcium calcium pulses chrome-extension://efaidnbmnnnibpcajpcglclefindmkaj/https://mcell.org/documentation/MCellQuickReferenceGuide.pdf release pattern interval</p>
<p>adding structural proteins to psd make drawing figures of what i expect, could do in future. t306 MT too. what would i expect?</p>
<p>effects of dephosphorylation could be interesting to look at:</p>
<p>In the GluN2B-bound state, the enzyme becomes resistant to the action of phosphatases (Cheriyan et al., 2011) indicating GluN2B-induced structural changes which can be abolished by specific mutations in CaMKII (Mayadevi et al., 2016). file:///C:/Users/susan/Downloads/fnmol-15-855752.pdf</p>
<p>we dont model this, but could be an emergent behaviour?</p>
<p>Importantly, the CaMKII/GluN2B complex was also shown to endure for hours even after dissociation of CaM from the kinase <span class="citation" data-cites="barcomb2016CaMKIIa">(<a href="#ref-barcomb2016CaMKIIa" role="doc-biblioref">Barcomb et al. 2016</a>)</span>.</p>
<p>discuss why biodynamo not used, the idea at the beginning and how it could be used in the future.</p>
<p>Because binding of CaMKII to the NMDA receptor is enhanced by CaMKII phosphorylation, the structurally mediated strengthening could be reversed by dephosphorylating CaMKII through the phosphatase-dependent processes (Fig. 4c,d) that are involved in DEPOTENTIATION73. https://www.nature.com/articles/nrn753 &lt;!– # limitations need to add a list</p>
<ul>
<li>could do with more runs, could do with further parameter analysis? # Results</li>
</ul>
<p>9.96181304^{-8} concentration of NMDAR/CAMKII complex at equilibrium (30 molecules converted to Molar) following equation in modelling chapter.</p>
<section id="camkii-dodecamer" class="level2" data-number="11.1">
<h2 data-number="11.1" class="anchored" data-anchor-id="camkii-dodecamer"><span class="header-section-number">11.1</span> CaMKII dodecamer</h2>
<p>mutant CaMKII/NMDAR binding abolition and how this replicates in the cBNGL model too.</p>
</section>
<section id="in-camkii286p-mt" class="level2" data-number="11.2">
<h2 data-number="11.2" class="anchored" data-anchor-id="in-camkii286p-mt"><span class="header-section-number">11.2</span> in camkii286p MT</h2>
<section id="does-this-model-reproduce-previously-suggested-results-of-camkii-activation" class="level3" data-number="11.2.1">
<h3 data-number="11.2.1" class="anchored" data-anchor-id="does-this-model-reproduce-previously-suggested-results-of-camkii-activation"><span class="header-section-number">11.2.1</span> Does this model reproduce previously suggested results of CaMKII activation?</h3>
</section>
<section id="camkii-interaction-with-nmdars" class="level3" data-number="11.2.2">
<h3 data-number="11.2.2" class="anchored" data-anchor-id="camkii-interaction-with-nmdars"><span class="header-section-number">11.2.2</span> CaMKII interaction with NMDARs</h3>
<p>Some conditions/hypotheses: Binding of cam_ca4_camkii is favoured to bind nmdar vs camkii_free, and camkii_p286 should also be favoured to bind nmdar. This is currently not included with how the model stands.</p>
<p>Yes it is included actually just not explicitly. Because camkii bound to camca4 is already more likely to exist in the open state; and so is camkii_p – so we expect these will be the ones binding more readily to nmdar (biology explanation because kinase domain is available to bind nmdars) - Do we need to specify different reaction rates? Or will camkii_camca4 bind more readily to nmdar anyway because camkii_camca4 is favoured to exist at higher quantities? How will this affect the model?</p>
<p>Cam_ca4 binding to camkii favours phosphorylation at t286 -&gt; this is also modelled but not expliclty. Camkii t306 P does not happen when camkii bound to nmdar (see condition reaction #8)</p>
<p>our results indicate that CaMKII binds with high affinity to NR2B/C when autophosphorylated on Thr286 independent of the presence of Ca2+ and CaM (Fig. 1)</p>
<p>does our model show that camkii_p, independent of cam binding, binds at higher rates? id expect camkii_p to be found bound to nmdar independently of cam_bound_camkii, so higher number of camkii_p_camfree, and less of camkii_p_cam_ca4</p>
<p>-&gt;model where camkii cannot p</p>
<p>CaMKII/GluN2B binding requires an initial Ca2+/CaM stimulus but then persists even after dissociation of CaM from the complex. https://www.jbc.org/article/S0021-9258(20)42434-2/fulltext</p>
<p>Both protein phosphatase 1 (PP1) and protein phosphatase 2A (PP2A) dephosphorylated pS1303 in the GluN2B CT [17]. In the PSD, PP1 is a major phosphatase catalyzing S1303 dephosphorylation [18]. https://pmc.ncbi.nlm.nih.gov/articles/PMC4435801/</p>
<ul>
<li><p>not checking for this but for future discussions: CaMKII binding to GluN2B even blocks this phosphorylation (Bayer et al., 2001). https://www.sciencedirect.com/science/article/pii/S0896627319304866</p></li>
<li><p>Secondary T305/T306 autophosphorylation takes place at ~100-fold slower rate than the T286/7 reaction, https://link.springer.com/chapter/10.1007/978-3-319-24364-1_3</p></li>
</ul>
</section>
</section>
</section>
<section id="same-as-before-most-activity-of-camkii-is-from-autonomous-not-due-to-bound-to-cam" class="level1" data-number="12">
<h1 data-number="12"><span class="header-section-number">12</span> same as before, most activity of camkii is from autonomous, not due to bound to cam:</h1>
<p>This is good stuff: CaMKIIα activation, but not CaMKIIα-CaM binding, integrates Ca2+ pulses. This suggests that most of the active CaMKIIα population is in a CaM-independent, autonomous activation state.</p>
<p>Taken together with our previous studies of CaMKIIα activation during repetitive Ca2+ pulses in the spine6,10, CaMKIIα activation, but not CaMKIIα-CaM binding, integrates Ca2+ pulses. This suggests that most of the active CaMKIIα population is in a CaM-independent, autonomous activation state. Our kinetic model also predicts that the CaMKIIα bound to CaM accounts for only a small fraction of CaMKIIα activity (~1/4), and most of the activity is from autonomous activation.</p>
<p>from paper: Mechanisms of Ca2+/CaM-dependent kinase II activation in single dendritic spines</p>
</section>
<section id="t305306-p" class="level1" data-number="13">
<h1 data-number="13"><span class="header-section-number">13</span> t305/306 p</h1>
<p>this is what we see in our model too: These results suggested that inhibitory phosphorylation at Thr305/Thr306 dynamically occurs during CaMKIIα activation, which inhibits the rebinding of Ca2+/CaM on CaMKIIα. However, preventing this regulation during the induction of sLTP (enhancing binding affinity to Ca2+/CaM) did not result in a higher level of Ca2+/CaM binding. https://link.springer.com/article/10.1038/s41467-019-10694-z?fromPaywallRec=false</p>
<p><strong>Aim 1.</strong> Develop a computational model of CaMKII/NMDAR interactions based on published models and known findings on CaMKII regulation. This way, the model itself functions as a testable hypothesis that synthesises literature knowledge about CaMKII/NMDAR interactions.</p>
<p><strong>Aim.</strong> Perform parameter sweeps for reaction rates that have not been measured experimentally. This allows mapping the model’s response to parameter changes and determining plausible ranges of the unknown parameters.</p>
<p><strong>Aim 2.</strong> Determine in detail how functional states of CaMKII contribute to stabilizing the CaMKII/NMDAR complex, defined as achieving long-lasting binding that reaches equilibrium. This includes investigating how conformational states of CaMKII and phosphorylation events influence the stability of the complex.</p>
<ul>
<li>our results indicate that CaMKII binds with high affinity to NR2B/C when autophosphorylated on Thr286 independent of the presence of Ca2+ and CaM (Fig. 1)</li>
</ul>
<p>does cam binding make a difference to camkii/nmdar binding</p>
<p>camkii_p free from cam binds more readily to nmdars than when camkii_p is bound to cam</p>
<p>does our model show that camkii_p, independent of cam binding, binds more readily than when camkii is free from cam binding? id expect camkii_p to be found bound to nmdar independently of cam_bound_camkii, so higher number of camkii_p_camfree, and less of camkii_p_cam_ca4</p>
<p><strong>Aim 3.</strong> Investigate how CaMKII/NMDAR binding influences CaMKII activity and function by examining an <em>in silico</em> mutation model where CaMKII-NMDAR interaction is disrupted.</p>
<p><strong>Aim 4.</strong> Investigate how CaMKII phosphorylation influences CaMKII/NMDAR binding by examining an <em>in silico</em> mutation model where CaMKII cannot be phosphorylated at key functional residues.</p>
<p><strong>Aim</strong> reproducibility. model follows fair principles. for details see chapter… <strong>aim</strong> ethics - data hazards.</p>
<section id="parameter-accuracy" class="level2" data-number="13.1">
<h2 data-number="13.1" class="anchored" data-anchor-id="parameter-accuracy"><span class="header-section-number">13.1</span> Parameter accuracy</h2>
</section>
</section>
<section id="trying-to-understand-how-parameters-are-limiting-or-not" class="level1" data-number="14">
<h1 data-number="14"><span class="header-section-number">14</span> Trying to understand how parameters are limiting or not</h1>
<div id="fig-diffparams" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-diffparams-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-diffparams" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-a" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-a-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="40-results-figures/2024-05-20_17-43-18_out.png" class="lightbox" data-gallery="fig-diffparams" title="Figure&nbsp;14.1&nbsp;(a): a"><img src="40-results-figures/2024-05-20_17-43-18_out.png" class="img-fluid figure-img" data-ref-parent="fig-diffparams"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-a-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) a
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-diffparams" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-b" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-b-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="40-results-figures/2024-05-22_16-05-00_out.png" class="lightbox" data-gallery="fig-diffparams" title="Figure&nbsp;14.1&nbsp;(b): b"><img src="40-results-figures/2024-05-22_16-05-00_out.png" class="img-fluid figure-img" data-ref-parent="fig-diffparams"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-b-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) b
</figcaption>
</figure>
</div>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-diffparams" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-c" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-c-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="40-results-figures/2024-05-22_16-16-27_out.png" class="lightbox" data-gallery="fig-diffparams" title="Figure&nbsp;14.1&nbsp;(c): c"><img src="40-results-figures/2024-05-22_16-16-27_out.png" class="img-fluid figure-img" data-ref-parent="fig-diffparams"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-c-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(c) c
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-diffparams" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-d" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-d-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="40-results-figures/2024-05-23_12-52-50_out.png" class="lightbox" data-gallery="fig-diffparams" title="Figure&nbsp;14.1&nbsp;(d): d"><img src="40-results-figures/2024-05-23_12-52-50_out.png" class="img-fluid figure-img" data-ref-parent="fig-diffparams"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-d-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(d) d
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-diffparams-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;14.1: Famous Elephants
</figcaption>
</figure>
</div>


<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-barcomb2016CaMKIIa" class="csl-entry" role="listitem">
Barcomb, Kelsey, Johannes W. Hell, Tim A. Benke, and K. Ulrich Bayer. 2016. <span>“The <span>CaMKII</span>/<span>GluN2B Protein Interaction Maintains Synaptic Strength</span> *.”</span> <em>Journal of Biological Chemistry</em> 291 (31): 16082–89. <a href="https://doi.org/10.1074/jbc.M116.734822">https://doi.org/10.1074/jbc.M116.734822</a>.
</div>
<div id="ref-chang2017CaMKII" class="csl-entry" role="listitem">
Chang, Jui-Yun, Paula Parra-Bueno, Tal Laviv, Erzsebet M. Szatmari, Seok-Jin R. Lee, and Ryohei Yasuda. 2017. <span>“<span>CaMKII Autophosphorylation Is Necessary</span> for <span>Optimal Integration</span> of <span>Ca2</span>+ <span>Signals</span> During <span>LTP Induction</span>, but <span>Not Maintenance</span>.”</span> <em>Neuron</em> 94 (4): 800–808.e4. <a href="https://doi.org/10.1016/j.neuron.2017.04.041">https://doi.org/10.1016/j.neuron.2017.04.041</a>.
</div>
<div id="ref-otmakhov2015Fast" class="csl-entry" role="listitem">
Otmakhov, Nikolai, Shaurav Regmi, and John E. Lisman. 2015. <span>“Fast <span>Decay</span> of <span>CaMKII FRET Sensor Signal</span> in <span>Spines</span> After <span>LTP Induction Is Not Due</span> to <span>Its Dephosphorylation</span>.”</span> <em>PLOS ONE</em> 10 (6): e0130457. <a href="https://doi.org/10.1371/journal.pone.0130457">https://doi.org/10.1371/journal.pone.0130457</a>.
</div>
</div>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./40-model-validation.html" class="pagination-link" aria-label="Validating model behaviour">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Validating model behaviour</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./publications.html" class="pagination-link" aria-label="List of publications and presentations resulting from this PhD">
        <span class="nav-page-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">List of publications and presentations resulting from this PhD</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>




</body></html>